{
 "cells": [
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Project: 2"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "import pandas as pd\n",
    "import numpy as np\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../Data/NCBI_Protein_seq_H5N1.csv'",
     "output_type": "error",
     "traceback": [
      "\u001b[1;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[1;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "Cell \u001b[1;32mIn[2], line 3\u001b[0m\n\u001b[0;32m      1\u001b[0m \u001b[38;5;66;03m# NCBI dataset with Protein sequence\u001b[39;00m\n\u001b[1;32m----> 3\u001b[0m df2\u001b[38;5;241m=\u001b[39m\u001b[43mpd\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mread_csv\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m../Data/NCBI_Protein_seq_H5N1.csv\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43mlow_memory\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\n\u001b[0;32m      4\u001b[0m df2_usa \u001b[38;5;241m=\u001b[39m df2[df2[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGeo_Location\u001b[39m\u001b[38;5;124m'\u001b[39m]\u001b[38;5;241m.\u001b[39mstr\u001b[38;5;241m.\u001b[39mcontains(\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mUSA\u001b[39m\u001b[38;5;124m'\u001b[39m, na\u001b[38;5;241m=\u001b[39m\u001b[38;5;28;01mFalse\u001b[39;00m)]\n\u001b[0;32m      5\u001b[0m df2_usa\u001b[38;5;241m=\u001b[39mdf2_usa\u001b[38;5;241m.\u001b[39mdrop(columns\u001b[38;5;241m=\u001b[39m[\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGenBank_RefSeq\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mSubmitters\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mGeo_Location\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mIsolate\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mTissue_Specimen_Source\u001b[39m\u001b[38;5;124m'\u001b[39m,\u001b[38;5;124m'\u001b[39m\u001b[38;5;124mRelease_Date\u001b[39m\u001b[38;5;124m'\u001b[39m]) \n",
      "File \u001b[1;32mc:\\Users\\Owner\\anaconda3\\envs\\dev\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1026\u001b[0m, in \u001b[0;36mread_csv\u001b[1;34m(filepath_or_buffer, sep, delimiter, header, names, index_col, usecols, dtype, engine, converters, true_values, false_values, skipinitialspace, skiprows, skipfooter, nrows, na_values, keep_default_na, na_filter, verbose, skip_blank_lines, parse_dates, infer_datetime_format, keep_date_col, date_parser, date_format, dayfirst, cache_dates, iterator, chunksize, compression, thousands, decimal, lineterminator, quotechar, quoting, doublequote, escapechar, comment, encoding, encoding_errors, dialect, on_bad_lines, delim_whitespace, low_memory, memory_map, float_precision, storage_options, dtype_backend)\u001b[0m\n\u001b[0;32m   1013\u001b[0m kwds_defaults \u001b[38;5;241m=\u001b[39m _refine_defaults_read(\n\u001b[0;32m   1014\u001b[0m     dialect,\n\u001b[0;32m   1015\u001b[0m     delimiter,\n\u001b[1;32m   (...)\u001b[0m\n\u001b[0;32m   1022\u001b[0m     dtype_backend\u001b[38;5;241m=\u001b[39mdtype_backend,\n\u001b[0;32m   1023\u001b[0m )\n\u001b[0;32m   1024\u001b[0m kwds\u001b[38;5;241m.\u001b[39mupdate(kwds_defaults)\n\u001b[1;32m-> 1026\u001b[0m \u001b[38;5;28;01mreturn\u001b[39;00m \u001b[43m_read\u001b[49m\u001b[43m(\u001b[49m\u001b[43mfilepath_or_buffer\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[43mkwds\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\Owner\\anaconda3\\envs\\dev\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:620\u001b[0m, in \u001b[0;36m_read\u001b[1;34m(filepath_or_buffer, kwds)\u001b[0m\n\u001b[0;32m    617\u001b[0m _validate_names(kwds\u001b[38;5;241m.\u001b[39mget(\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mnames\u001b[39m\u001b[38;5;124m\"\u001b[39m, \u001b[38;5;28;01mNone\u001b[39;00m))\n\u001b[0;32m    619\u001b[0m \u001b[38;5;66;03m# Create the parser.\u001b[39;00m\n\u001b[1;32m--> 620\u001b[0m parser \u001b[38;5;241m=\u001b[39m TextFileReader(filepath_or_buffer, \u001b[38;5;241m*\u001b[39m\u001b[38;5;241m*\u001b[39mkwds)\n\u001b[0;32m    622\u001b[0m \u001b[38;5;28;01mif\u001b[39;00m chunksize \u001b[38;5;129;01mor\u001b[39;00m iterator:\n\u001b[0;32m    623\u001b[0m     \u001b[38;5;28;01mreturn\u001b[39;00m parser\n",
      "File \u001b[1;32mc:\\Users\\Owner\\anaconda3\\envs\\dev\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1620\u001b[0m, in \u001b[0;36mTextFileReader.__init__\u001b[1;34m(self, f, engine, **kwds)\u001b[0m\n\u001b[0;32m   1617\u001b[0m     \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39moptions[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m] \u001b[38;5;241m=\u001b[39m kwds[\u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mhas_index_names\u001b[39m\u001b[38;5;124m\"\u001b[39m]\n\u001b[0;32m   1619\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles: IOHandles \u001b[38;5;241m|\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[1;32m-> 1620\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39m_engine \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43m_make_engine\u001b[49m\u001b[43m(\u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mengine\u001b[49m\u001b[43m)\u001b[49m\n",
      "File \u001b[1;32mc:\\Users\\Owner\\anaconda3\\envs\\dev\\lib\\site-packages\\pandas\\io\\parsers\\readers.py:1880\u001b[0m, in \u001b[0;36mTextFileReader._make_engine\u001b[1;34m(self, f, engine)\u001b[0m\n\u001b[0;32m   1878\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m mode:\n\u001b[0;32m   1879\u001b[0m         mode \u001b[38;5;241m+\u001b[39m\u001b[38;5;241m=\u001b[39m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m\n\u001b[1;32m-> 1880\u001b[0m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;241m=\u001b[39m \u001b[43mget_handle\u001b[49m\u001b[43m(\u001b[49m\n\u001b[0;32m   1881\u001b[0m \u001b[43m    \u001b[49m\u001b[43mf\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1882\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1883\u001b[0m \u001b[43m    \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1884\u001b[0m \u001b[43m    \u001b[49m\u001b[43mcompression\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mcompression\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1885\u001b[0m \u001b[43m    \u001b[49m\u001b[43mmemory_map\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mmemory_map\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mFalse\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1886\u001b[0m \u001b[43m    \u001b[49m\u001b[43mis_text\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mis_text\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1887\u001b[0m \u001b[43m    \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mencoding_errors\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstrict\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1888\u001b[0m \u001b[43m    \u001b[49m\u001b[43mstorage_options\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;28;43mself\u001b[39;49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43moptions\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mget\u001b[49m\u001b[43m(\u001b[49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43mstorage_options\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\u001b[43m \u001b[49m\u001b[38;5;28;43;01mNone\u001b[39;49;00m\u001b[43m)\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m   1889\u001b[0m \u001b[43m\u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m   1890\u001b[0m \u001b[38;5;28;01massert\u001b[39;00m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles \u001b[38;5;129;01mis\u001b[39;00m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;28;01mNone\u001b[39;00m\n\u001b[0;32m   1891\u001b[0m f \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mself\u001b[39m\u001b[38;5;241m.\u001b[39mhandles\u001b[38;5;241m.\u001b[39mhandle\n",
      "File \u001b[1;32mc:\\Users\\Owner\\anaconda3\\envs\\dev\\lib\\site-packages\\pandas\\io\\common.py:873\u001b[0m, in \u001b[0;36mget_handle\u001b[1;34m(path_or_buf, mode, encoding, compression, memory_map, is_text, errors, storage_options)\u001b[0m\n\u001b[0;32m    868\u001b[0m \u001b[38;5;28;01melif\u001b[39;00m \u001b[38;5;28misinstance\u001b[39m(handle, \u001b[38;5;28mstr\u001b[39m):\n\u001b[0;32m    869\u001b[0m     \u001b[38;5;66;03m# Check whether the filename is to be opened in binary mode.\u001b[39;00m\n\u001b[0;32m    870\u001b[0m     \u001b[38;5;66;03m# Binary mode does not support 'encoding' and 'newline'.\u001b[39;00m\n\u001b[0;32m    871\u001b[0m     \u001b[38;5;28;01mif\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mencoding \u001b[38;5;129;01mand\u001b[39;00m \u001b[38;5;124m\"\u001b[39m\u001b[38;5;124mb\u001b[39m\u001b[38;5;124m\"\u001b[39m \u001b[38;5;129;01mnot\u001b[39;00m \u001b[38;5;129;01min\u001b[39;00m ioargs\u001b[38;5;241m.\u001b[39mmode:\n\u001b[0;32m    872\u001b[0m         \u001b[38;5;66;03m# Encoding\u001b[39;00m\n\u001b[1;32m--> 873\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28;43mopen\u001b[39;49m\u001b[43m(\u001b[49m\n\u001b[0;32m    874\u001b[0m \u001b[43m            \u001b[49m\u001b[43mhandle\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    875\u001b[0m \u001b[43m            \u001b[49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mmode\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    876\u001b[0m \u001b[43m            \u001b[49m\u001b[43mencoding\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43mioargs\u001b[49m\u001b[38;5;241;43m.\u001b[39;49m\u001b[43mencoding\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    877\u001b[0m \u001b[43m            \u001b[49m\u001b[43merrors\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[43merrors\u001b[49m\u001b[43m,\u001b[49m\n\u001b[0;32m    878\u001b[0m \u001b[43m            \u001b[49m\u001b[43mnewline\u001b[49m\u001b[38;5;241;43m=\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[38;5;124;43m\"\u001b[39;49m\u001b[43m,\u001b[49m\n\u001b[0;32m    879\u001b[0m \u001b[43m        \u001b[49m\u001b[43m)\u001b[49m\n\u001b[0;32m    880\u001b[0m     \u001b[38;5;28;01melse\u001b[39;00m:\n\u001b[0;32m    881\u001b[0m         \u001b[38;5;66;03m# Binary mode\u001b[39;00m\n\u001b[0;32m    882\u001b[0m         handle \u001b[38;5;241m=\u001b[39m \u001b[38;5;28mopen\u001b[39m(handle, ioargs\u001b[38;5;241m.\u001b[39mmode)\n",
      "\u001b[1;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../Data/NCBI_Protein_seq_H5N1.csv'"
     ]
    }
   ],
   "source": [
    "# NCBI dataset with Protein sequence\n",
    "\n",
    "df2=pd.read_csv(\"../Data/NCBI_Protein_seq_H5N1.csv\",low_memory=False)\n",
    "df2_usa = df2[df2['Geo_Location'].str.contains('USA', na=False)]\n",
    "df2_usa=df2_usa.drop(columns=['GenBank_RefSeq','Submitters','Geo_Location','Isolate','Tissue_Specimen_Source','Release_Date']) \n",
    "df2_usa.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Other Names</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Serovar</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>...</th>\n",
       "      <th>GC Content</th>\n",
       "      <th>Collection Year</th>\n",
       "      <th>Geographic Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Name</th>\n",
       "      <th>Host Common Name</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Lab Host</th>\n",
       "      <th>Date Inserted</th>\n",
       "      <th>Date Modified</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>44.396920</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:28:55.112Z</td>\n",
       "      <td>2024-02-24T00:28:55.112Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>43.761220</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:28:55.272Z</td>\n",
       "      <td>2024-02-24T00:28:55.272Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>43.921234</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:28:55.190Z</td>\n",
       "      <td>2024-02-24T00:28:55.190Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11320.50625</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>41.548897</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:29:01.510Z</td>\n",
       "      <td>2024-02-24T00:29:01.510Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11320.50625</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>5</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>47.119080</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:28:55.365Z</td>\n",
       "      <td>2024-02-24T00:28:55.365Z</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Genome ID                                        Genome Name  \\\n",
       "0  11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "1  11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "2  11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "3  11320.50625  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "4  11320.50625  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "\n",
       "   Other Names            Family                Genus  \\\n",
       "0          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "1          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "2          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "3          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "4          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "\n",
       "                          Species                               Strain  \\\n",
       "0  Alphainfluenzavirus influenzae  A/mute swan/Mangystau/1-S24R-2/2024   \n",
       "1  Alphainfluenzavirus influenzae  A/mute swan/Mangystau/1-S24R-2/2024   \n",
       "2  Alphainfluenzavirus influenzae  A/mute swan/Mangystau/1-S24R-2/2024   \n",
       "3  Alphainfluenzavirus influenzae  A/mute swan/Mangystau/1-S24R-2/2024   \n",
       "4  Alphainfluenzavirus influenzae  A/mute swan/Mangystau/1-S24R-2/2024   \n",
       "\n",
       "  Serovar  Segment Subtype  ... GC Content  Collection Year Geographic Group  \\\n",
       "0    H5N1        1    H5N1  ...  44.396920             2024             Asia   \n",
       "1    H5N1        3    H5N1  ...  43.761220             2024             Asia   \n",
       "2    H5N1        2    H5N1  ...  43.921234             2024             Asia   \n",
       "3    H5N1        4    H5N1  ...  41.548897             2024             Asia   \n",
       "4    H5N1        5    H5N1  ...  47.119080             2024             Asia   \n",
       "\n",
       "     Geographic Location  Host Name Host Common Name  Host Group Lab Host  \\\n",
       "0  Kazakhstan: Mangystau  Mute swan        Mute Swan       Avian      NaN   \n",
       "1  Kazakhstan: Mangystau  Mute swan        Mute Swan       Avian      NaN   \n",
       "2  Kazakhstan: Mangystau  Mute swan        Mute Swan       Avian      NaN   \n",
       "3  Kazakhstan: Mangystau  Mute swan        Mute Swan       Avian      NaN   \n",
       "4  Kazakhstan: Mangystau  Mute swan        Mute Swan       Avian      NaN   \n",
       "\n",
       "              Date Inserted             Date Modified  \n",
       "0  2024-02-24T00:28:55.112Z  2024-02-24T00:28:55.112Z  \n",
       "1  2024-02-24T00:28:55.272Z  2024-02-24T00:28:55.272Z  \n",
       "2  2024-02-24T00:28:55.190Z  2024-02-24T00:28:55.190Z  \n",
       "3  2024-02-24T00:29:01.510Z  2024-02-24T00:29:01.510Z  \n",
       "4  2024-02-24T00:28:55.365Z  2024-02-24T00:28:55.365Z  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 198,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# BV-BRC dataset from GenBank\n",
    "df3 = pd.read_csv(\"../Data/H5N1_Outbreak_Genome_GenBank.csv\", low_memory=False)\n",
    "df3.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Other Names</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Serovar</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>...</th>\n",
       "      <th>GC Content</th>\n",
       "      <th>Collection Date</th>\n",
       "      <th>Collection Year</th>\n",
       "      <th>Geographic Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Name</th>\n",
       "      <th>Host Common Name</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Date Inserted</th>\n",
       "      <th>Date Modified</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11320.58840</td>\n",
       "      <td>Influenza A virus A/Cattle/USA/24-008766-001-v...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cattle/USA/24-008766-001-v/2024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>4</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>41.784040</td>\n",
       "      <td>2024</td>\n",
       "      <td>2024</td>\n",
       "      <td>North America</td>\n",
       "      <td>USA</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Nonhuman Mammal</td>\n",
       "      <td>2024-06-09T21:02:30.513Z</td>\n",
       "      <td>2024-06-09T21:02:30.513Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>11320.58840</td>\n",
       "      <td>Influenza A virus A/Cattle/USA/24-008766-001-v...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cattle/USA/24-008766-001-v/2024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>6</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>43.900710</td>\n",
       "      <td>2024</td>\n",
       "      <td>2024</td>\n",
       "      <td>North America</td>\n",
       "      <td>USA</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Nonhuman Mammal</td>\n",
       "      <td>2024-06-09T21:02:30.676Z</td>\n",
       "      <td>2024-06-09T21:02:30.676Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>11320.58840</td>\n",
       "      <td>Influenza A virus A/Cattle/USA/24-008766-001-v...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cattle/USA/24-008766-001-v/2024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>5</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>47.561790</td>\n",
       "      <td>2024</td>\n",
       "      <td>2024</td>\n",
       "      <td>North America</td>\n",
       "      <td>USA</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Nonhuman Mammal</td>\n",
       "      <td>2024-06-09T21:02:30.191Z</td>\n",
       "      <td>2024-06-09T21:02:30.191Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11320.58840</td>\n",
       "      <td>Influenza A virus A/Cattle/USA/24-008766-001-v...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cattle/USA/24-008766-001-v/2024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>7</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>49.694500</td>\n",
       "      <td>2024</td>\n",
       "      <td>2024</td>\n",
       "      <td>North America</td>\n",
       "      <td>USA</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Nonhuman Mammal</td>\n",
       "      <td>2024-06-09T21:02:30.430Z</td>\n",
       "      <td>2024-06-09T21:02:30.430Z</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>11320.58841</td>\n",
       "      <td>Influenza A virus A/Cattle/USA/24-008766-001-v...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cattle/USA/24-008766-001-v/2024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>3</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>...</td>\n",
       "      <td>44.211994</td>\n",
       "      <td>2024</td>\n",
       "      <td>2024</td>\n",
       "      <td>North America</td>\n",
       "      <td>USA</td>\n",
       "      <td>Cattle</td>\n",
       "      <td>Cow</td>\n",
       "      <td>Nonhuman Mammal</td>\n",
       "      <td>2024-06-09T21:02:30.351Z</td>\n",
       "      <td>2024-06-09T21:02:30.351Z</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 30 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "     Genome ID                                        Genome Name  \\\n",
       "0  11320.58840  Influenza A virus A/Cattle/USA/24-008766-001-v...   \n",
       "1  11320.58840  Influenza A virus A/Cattle/USA/24-008766-001-v...   \n",
       "2  11320.58840  Influenza A virus A/Cattle/USA/24-008766-001-v...   \n",
       "3  11320.58840  Influenza A virus A/Cattle/USA/24-008766-001-v...   \n",
       "4  11320.58841  Influenza A virus A/Cattle/USA/24-008766-001-v...   \n",
       "\n",
       "   Other Names            Family                Genus  \\\n",
       "0          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "1          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "2          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "3          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "4          NaN  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "\n",
       "                          Species                             Strain  Serovar  \\\n",
       "0  Alphainfluenzavirus influenzae  A/Cattle/USA/24-008766-001-v/2024      NaN   \n",
       "1  Alphainfluenzavirus influenzae  A/Cattle/USA/24-008766-001-v/2024      NaN   \n",
       "2  Alphainfluenzavirus influenzae  A/Cattle/USA/24-008766-001-v/2024      NaN   \n",
       "3  Alphainfluenzavirus influenzae  A/Cattle/USA/24-008766-001-v/2024      NaN   \n",
       "4  Alphainfluenzavirus influenzae  A/Cattle/USA/24-008766-001-v/2024      NaN   \n",
       "\n",
       "   Segment Subtype  ... GC Content  Collection Date Collection Year  \\\n",
       "0        4    H5N1  ...  41.784040             2024            2024   \n",
       "1        6    H5N1  ...  43.900710             2024            2024   \n",
       "2        5    H5N1  ...  47.561790             2024            2024   \n",
       "3        7    H5N1  ...  49.694500             2024            2024   \n",
       "4        3    H5N1  ...  44.211994             2024            2024   \n",
       "\n",
       "   Geographic Group Geographic Location Host Name  Host Common Name  \\\n",
       "0     North America                 USA    Cattle               Cow   \n",
       "1     North America                 USA    Cattle               Cow   \n",
       "2     North America                 USA    Cattle               Cow   \n",
       "3     North America                 USA    Cattle               Cow   \n",
       "4     North America                 USA    Cattle               Cow   \n",
       "\n",
       "        Host Group             Date Inserted             Date Modified  \n",
       "0  Nonhuman Mammal  2024-06-09T21:02:30.513Z  2024-06-09T21:02:30.513Z  \n",
       "1  Nonhuman Mammal  2024-06-09T21:02:30.676Z  2024-06-09T21:02:30.676Z  \n",
       "2  Nonhuman Mammal  2024-06-09T21:02:30.191Z  2024-06-09T21:02:30.191Z  \n",
       "3  Nonhuman Mammal  2024-06-09T21:02:30.430Z  2024-06-09T21:02:30.430Z  \n",
       "4  Nonhuman Mammal  2024-06-09T21:02:30.351Z  2024-06-09T21:02:30.351Z  \n",
       "\n",
       "[5 rows x 30 columns]"
      ]
     },
     "execution_count": 199,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#BV-BRC dataset from SRA database\n",
    "df4 = pd.read_csv(\"../Data/H5N1_Outbreak_Genome_SRA_database.csv\",low_memory=False)\n",
    "df4.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 36531 entries, 0 to 36530\n",
      "Data columns (total 30 columns):\n",
      " #   Column                Non-Null Count  Dtype  \n",
      "---  ------                --------------  -----  \n",
      " 0   Genome ID             36531 non-null  float64\n",
      " 1   Genome Name           36531 non-null  object \n",
      " 2   Other Names           0 non-null      float64\n",
      " 3   Family                36531 non-null  object \n",
      " 4   Genus                 36531 non-null  object \n",
      " 5   Species               36531 non-null  object \n",
      " 6   Strain                36531 non-null  object \n",
      " 7   Serovar               23549 non-null  object \n",
      " 8   Segment               36531 non-null  int64  \n",
      " 9   Subtype               36531 non-null  object \n",
      " 10  H5 Clade              30431 non-null  object \n",
      " 11  Clade                 0 non-null      float64\n",
      " 12  Subclade              27771 non-null  object \n",
      " 13  Completion Date       23549 non-null  object \n",
      " 14  BioProject Accession  35108 non-null  object \n",
      " 15  BioSample Accession   34272 non-null  object \n",
      " 16  Assembly Accession    0 non-null      float64\n",
      " 17  SRA Accession         32432 non-null  object \n",
      " 18  GenBank Accessions    23549 non-null  object \n",
      " 19  Size                  36531 non-null  int64  \n",
      " 20  GC Content            36531 non-null  float64\n",
      " 21  Collection Year       36531 non-null  int64  \n",
      " 22  Geographic Group      24603 non-null  object \n",
      " 23  Geographic Location   36531 non-null  object \n",
      " 24  Host Name             36531 non-null  object \n",
      " 25  Host Common Name      35917 non-null  object \n",
      " 26  Host Group            35917 non-null  object \n",
      " 27  Lab Host              16 non-null     object \n",
      " 28  Date Inserted         36531 non-null  object \n",
      " 29  Date Modified         36531 non-null  object \n",
      "dtypes: float64(5), int64(3), object(22)\n",
      "memory usage: 8.4+ MB\n"
     ]
    }
   ],
   "source": [
    "# Genbank database\n",
    "\n",
    "df3.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 10431 entries, 0 to 10430\n",
      "Data columns (total 30 columns):\n",
      " #   Column                Non-Null Count  Dtype  \n",
      "---  ------                --------------  -----  \n",
      " 0   Genome ID             10431 non-null  float64\n",
      " 1   Genome Name           10431 non-null  object \n",
      " 2   Other Names           0 non-null      float64\n",
      " 3   Family                10431 non-null  object \n",
      " 4   Genus                 10431 non-null  object \n",
      " 5   Species               10431 non-null  object \n",
      " 6   Strain                10431 non-null  object \n",
      " 7   Serovar               0 non-null      float64\n",
      " 8   Segment               10431 non-null  int64  \n",
      " 9   Subtype               10431 non-null  object \n",
      " 10  H5 Clade              7694 non-null   object \n",
      " 11  Clade                 0 non-null      float64\n",
      " 12  Subclade              10305 non-null  object \n",
      " 13  Completion Date       0 non-null      float64\n",
      " 14  BioProject Accession  10431 non-null  object \n",
      " 15  BioSample Accession   10431 non-null  object \n",
      " 16  Assembly Accession    0 non-null      float64\n",
      " 17  SRA Accession         10431 non-null  object \n",
      " 18  GenBank Accessions    0 non-null      float64\n",
      " 19  Size                  10431 non-null  int64  \n",
      " 20  GC Content            10431 non-null  float64\n",
      " 21  Collection Date       10431 non-null  object \n",
      " 22  Collection Year       10431 non-null  int64  \n",
      " 23  Geographic Group      10431 non-null  object \n",
      " 24  Geographic Location   10431 non-null  object \n",
      " 25  Host Name             10174 non-null  object \n",
      " 26  Host Common Name      10150 non-null  object \n",
      " 27  Host Group            10150 non-null  object \n",
      " 28  Date Inserted         10431 non-null  object \n",
      " 29  Date Modified         10431 non-null  object \n",
      "dtypes: float64(8), int64(3), object(19)\n",
      "memory usage: 2.4+ MB\n"
     ]
    }
   ],
   "source": [
    "#SRA database\n",
    "df4.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Cleaning"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "RangeIndex: 46962 entries, 0 to 46961\n",
      "Data columns (total 28 columns):\n",
      " #   Column                Non-Null Count  Dtype  \n",
      "---  ------                --------------  -----  \n",
      " 0   Genome ID             46962 non-null  float64\n",
      " 1   Genome Name           46962 non-null  object \n",
      " 2   Family                46962 non-null  object \n",
      " 3   Genus                 46962 non-null  object \n",
      " 4   Species               46962 non-null  object \n",
      " 5   Strain                46962 non-null  object \n",
      " 6   Serovar               23549 non-null  object \n",
      " 7   Segment               46962 non-null  int64  \n",
      " 8   Subtype               46962 non-null  object \n",
      " 9   H5 Clade              38125 non-null  object \n",
      " 10  Subclade              38076 non-null  object \n",
      " 11  Completion Date       23549 non-null  object \n",
      " 12  BioProject Accession  45539 non-null  object \n",
      " 13  BioSample Accession   44703 non-null  object \n",
      " 14  SRA Accession         42863 non-null  object \n",
      " 15  GenBank Accessions    23549 non-null  object \n",
      " 16  Size                  46962 non-null  int64  \n",
      " 17  GC Content            46962 non-null  float64\n",
      " 18  Collection Year       46962 non-null  int64  \n",
      " 19  Geographic Group      35034 non-null  object \n",
      " 20  Geographic Location   46962 non-null  object \n",
      " 21  Host Name             46705 non-null  object \n",
      " 22  Host Common Name      46067 non-null  object \n",
      " 23  Host Group            46067 non-null  object \n",
      " 24  Lab Host              16 non-null     object \n",
      " 25  Date Inserted         46962 non-null  object \n",
      " 26  Date Modified         46962 non-null  object \n",
      " 27  Collection Date       10431 non-null  object \n",
      "dtypes: float64(2), int64(3), object(23)\n",
      "memory usage: 10.0+ MB\n"
     ]
    }
   ],
   "source": [
    "# Concat the H5N1 Outbreak data from GenBank and SRA Databases\n",
    "bv_brc_df=pd.concat([df3,df4],axis=0, ignore_index=True)\n",
    "\n",
    "# Drop the columns with all null values\n",
    "bv_brc_df.dropna(axis=1, how='all', inplace=True)\n",
    "\n",
    "bv_brc_df.info()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Serovar</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>...</th>\n",
       "      <th>Collection Year</th>\n",
       "      <th>Geographic Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Name</th>\n",
       "      <th>Host Common Name</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Lab Host</th>\n",
       "      <th>Date Inserted</th>\n",
       "      <th>Date Modified</th>\n",
       "      <th>Collection Date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:28:55.112Z</td>\n",
       "      <td>2024-02-24T00:28:55.112Z</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>11320.50625</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>...</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute swan</td>\n",
       "      <td>Mute Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-02-24T00:29:01.510Z</td>\n",
       "      <td>2024-02-24T00:29:01.510Z</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>11320.52640</td>\n",
       "      <td>Influenza A virus A/Cygnus cygnus/Karakol lake...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cygnus cygnus/Karakol lake/01/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau region, Umirzak village,...</td>\n",
       "      <td>Cygnus cygnus</td>\n",
       "      <td>Whooper Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-21T19:03:21.509Z</td>\n",
       "      <td>2024-03-21T19:03:21.509Z</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>11320.52641</td>\n",
       "      <td>Influenza A virus A/Cygnus cygnus/Karakol lake...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Cygnus cygnus/Karakol lake/01/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2024</td>\n",
       "      <td>Asia</td>\n",
       "      <td>Kazakhstan: Mangystau region, Umirzak village,...</td>\n",
       "      <td>Cygnus cygnus</td>\n",
       "      <td>Whooper Swan</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-03-21T19:03:21.422Z</td>\n",
       "      <td>2024-03-21T19:03:21.422Z</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>11320.54589</td>\n",
       "      <td>Influenza A virus A/hooded merganser/South Car...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>2024</td>\n",
       "      <td>North America</td>\n",
       "      <td>USA</td>\n",
       "      <td>Lophodytes cucullatus</td>\n",
       "      <td>Hooded Merganser</td>\n",
       "      <td>Avian</td>\n",
       "      <td>NaN</td>\n",
       "      <td>2024-04-22T04:03:07.283Z</td>\n",
       "      <td>2024-04-22T04:03:07.283Z</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 28 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Genome ID                                        Genome Name  \\\n",
       "0   11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "3   11320.50625  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "8   11320.52640  Influenza A virus A/Cygnus cygnus/Karakol lake...   \n",
       "10  11320.52641  Influenza A virus A/Cygnus cygnus/Karakol lake...   \n",
       "16  11320.54589  Influenza A virus A/hooded merganser/South Car...   \n",
       "\n",
       "              Family                Genus                         Species  \\\n",
       "0   Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "3   Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "8   Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "10  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "16  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                            Strain Serovar  Segment Subtype  \\\n",
       "0              A/mute swan/Mangystau/1-S24R-2/2024    H5N1        1    H5N1   \n",
       "3              A/mute swan/Mangystau/1-S24R-2/2024    H5N1        4    H5N1   \n",
       "8             A/Cygnus cygnus/Karakol lake/01/2024    H5N1        1    H5N1   \n",
       "10            A/Cygnus cygnus/Karakol lake/01/2024    H5N1        3    H5N1   \n",
       "16  A/hooded merganser/South Carolina/W24-028/2024    H5N1        8    H5N1   \n",
       "\n",
       "    H5 Clade  ... Collection Year Geographic Group  \\\n",
       "0        NaN  ...            2024             Asia   \n",
       "3   2.3.4.4b  ...            2024             Asia   \n",
       "8        NaN  ...            2024             Asia   \n",
       "10       NaN  ...            2024             Asia   \n",
       "16       NaN  ...            2024    North America   \n",
       "\n",
       "                                  Geographic Location              Host Name  \\\n",
       "0                               Kazakhstan: Mangystau              Mute swan   \n",
       "3                               Kazakhstan: Mangystau              Mute swan   \n",
       "8   Kazakhstan: Mangystau region, Umirzak village,...          Cygnus cygnus   \n",
       "10  Kazakhstan: Mangystau region, Umirzak village,...          Cygnus cygnus   \n",
       "16                                                USA  Lophodytes cucullatus   \n",
       "\n",
       "    Host Common Name Host Group  Lab Host             Date Inserted  \\\n",
       "0          Mute Swan      Avian       NaN  2024-02-24T00:28:55.112Z   \n",
       "3          Mute Swan      Avian       NaN  2024-02-24T00:29:01.510Z   \n",
       "8       Whooper Swan      Avian       NaN  2024-03-21T19:03:21.509Z   \n",
       "10      Whooper Swan      Avian       NaN  2024-03-21T19:03:21.422Z   \n",
       "16  Hooded Merganser      Avian       NaN  2024-04-22T04:03:07.283Z   \n",
       "\n",
       "               Date Modified Collection Date  \n",
       "0   2024-02-24T00:28:55.112Z             NaN  \n",
       "3   2024-02-24T00:29:01.510Z             NaN  \n",
       "8   2024-03-21T19:03:21.509Z             NaN  \n",
       "10  2024-03-21T19:03:21.422Z             NaN  \n",
       "16  2024-04-22T04:03:07.283Z             NaN  \n",
       "\n",
       "[5 rows x 28 columns]"
      ]
     },
     "execution_count": 203,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the duplicates in the concatenated dataframe for the 'Genome ID' values\n",
    "bv_brc_df.drop_duplicates(subset=['Genome ID'], keep='first', inplace=True)\n",
    "\n",
    "# Rename the 'SRA Accession\" column to match with other dataset\n",
    "bv_brc_df=bv_brc_df.rename(columns={'SRA Accession':'SRA_Accession'})\n",
    "\n",
    "bv_brc_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 4897 entries, 0 to 46955\n",
      "Data columns (total 28 columns):\n",
      " #   Column                Non-Null Count  Dtype  \n",
      "---  ------                --------------  -----  \n",
      " 0   Genome ID             4897 non-null   float64\n",
      " 1   Genome Name           4897 non-null   object \n",
      " 2   Family                4897 non-null   object \n",
      " 3   Genus                 4897 non-null   object \n",
      " 4   Species               4897 non-null   object \n",
      " 5   Strain                4897 non-null   object \n",
      " 6   Serovar               2475 non-null   object \n",
      " 7   Segment               4897 non-null   int64  \n",
      " 8   Subtype               4897 non-null   object \n",
      " 9   H5 Clade              3919 non-null   object \n",
      " 10  Subclade              3968 non-null   object \n",
      " 11  Completion Date       2475 non-null   object \n",
      " 12  BioProject Accession  4723 non-null   object \n",
      " 13  BioSample Accession   4598 non-null   object \n",
      " 14  SRA_Accession         4401 non-null   object \n",
      " 15  GenBank Accessions    2475 non-null   object \n",
      " 16  Size                  4897 non-null   int64  \n",
      " 17  GC Content            4897 non-null   float64\n",
      " 18  Collection Year       4897 non-null   int64  \n",
      " 19  Geographic Group      3676 non-null   object \n",
      " 20  Geographic Location   4897 non-null   object \n",
      " 21  Host Name             4873 non-null   object \n",
      " 22  Host Common Name      4792 non-null   object \n",
      " 23  Host Group            4792 non-null   object \n",
      " 24  Lab Host              3 non-null      object \n",
      " 25  Date Inserted         4897 non-null   object \n",
      " 26  Date Modified         4897 non-null   object \n",
      " 27  Collection Date       1116 non-null   object \n",
      "dtypes: float64(2), int64(3), object(23)\n",
      "memory usage: 1.1+ MB\n"
     ]
    }
   ],
   "source": [
    "# After removing duplicates\n",
    "bv_brc_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 89130 entries, 0 to 138112\n",
      "Data columns (total 44 columns):\n",
      " #   Column                Non-Null Count  Dtype  \n",
      "---  ------                --------------  -----  \n",
      " 0   Genome ID             4897 non-null   float64\n",
      " 1   Genome Name           4897 non-null   object \n",
      " 2   Family                4897 non-null   object \n",
      " 3   Genus                 4897 non-null   object \n",
      " 4   Species               4897 non-null   object \n",
      " 5   Strain                4897 non-null   object \n",
      " 6   Serovar               2475 non-null   object \n",
      " 7   Segment               4897 non-null   float64\n",
      " 8   Subtype               4897 non-null   object \n",
      " 9   H5 Clade              3919 non-null   object \n",
      " 10  Subclade              3968 non-null   object \n",
      " 11  Completion Date       2475 non-null   object \n",
      " 12  BioProject Accession  4723 non-null   object \n",
      " 13  BioSample Accession   4598 non-null   object \n",
      " 14  SRA_Accession         4401 non-null   object \n",
      " 15  GenBank Accessions    2475 non-null   object \n",
      " 16  Size                  4897 non-null   float64\n",
      " 17  GC Content            4897 non-null   float64\n",
      " 18  Collection Year       4897 non-null   float64\n",
      " 19  Geographic Group      3676 non-null   object \n",
      " 20  Geographic Location   4897 non-null   object \n",
      " 21  Host Name             4873 non-null   object \n",
      " 22  Host Common Name      4792 non-null   object \n",
      " 23  Host Group            4792 non-null   object \n",
      " 24  Lab Host              3 non-null      object \n",
      " 25  Date Inserted         4897 non-null   object \n",
      " 26  Date Modified         4897 non-null   object \n",
      " 27  Collection Date       1116 non-null   object \n",
      " 28  Accession             89084 non-null  object \n",
      " 29  Organism_Name         89084 non-null  object \n",
      " 30  Assembly              81825 non-null  object \n",
      " 31  Nucleotide            89084 non-null  object \n",
      " 32  SRA_Accession         33116 non-null  object \n",
      " 33  Species               89084 non-null  object \n",
      " 34  Genus                 89084 non-null  object \n",
      " 35  Family                89084 non-null  object \n",
      " 36  Molecule_type         89084 non-null  object \n",
      " 37  Length                89084 non-null  float64\n",
      " 38  Genotype              89084 non-null  object \n",
      " 39  Segment               89074 non-null  object \n",
      " 40  Protein               89084 non-null  object \n",
      " 41  USA                   79151 non-null  object \n",
      " 42  Host                  84746 non-null  object \n",
      " 43  Collection_Date       89074 non-null  object \n",
      "dtypes: float64(6), object(38)\n",
      "memory usage: 30.6+ MB\n"
     ]
    }
   ],
   "source": [
    "# Concatenate BV-BRC dataset and the NCBI protein seq dataset\n",
    "\n",
    "h5n1_records_df = pd.concat([bv_brc_df,df2_usa],axis=1)\n",
    "h5n1_records_df.info()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Serovar</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>...</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Family</th>\n",
       "      <th>Molecule_type</th>\n",
       "      <th>Length</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Protein</th>\n",
       "      <th>USA</th>\n",
       "      <th>Host</th>\n",
       "      <th>Collection_Date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>11320.54589</td>\n",
       "      <td>Influenza A virus A/hooded merganser/South Car...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>252.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3</td>\n",
       "      <td>PA-X protein</td>\n",
       "      <td>CA</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>12/9/2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>11320.54595</td>\n",
       "      <td>Influenza A virus A/wood duck/North Carolina/W...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>90.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2</td>\n",
       "      <td>PB1-F2 protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>11320.54640</td>\n",
       "      <td>Influenza A virus A/Texas/37/2024</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>90.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2</td>\n",
       "      <td>PB1-F2 protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>11320.57078</td>\n",
       "      <td>Influenza A virus A/canada_goose/Indiana/24-00...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>...</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>252.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3</td>\n",
       "      <td>PA-X protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 44 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Genome ID                                        Genome Name  \\\n",
       "0   11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "16  11320.54589  Influenza A virus A/hooded merganser/South Car...   \n",
       "26  11320.54595  Influenza A virus A/wood duck/North Carolina/W...   \n",
       "38  11320.54640                  Influenza A virus A/Texas/37/2024   \n",
       "40  11320.57078  Influenza A virus A/canada_goose/Indiana/24-00...   \n",
       "\n",
       "              Family                Genus                         Species  \\\n",
       "0   Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "16  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "26  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "38  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "40  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                            Strain Serovar Segment Subtype  \\\n",
       "0              A/mute swan/Mangystau/1-S24R-2/2024    H5N1     1.0    H5N1   \n",
       "16  A/hooded merganser/South Carolina/W24-028/2024    H5N1     8.0    H5N1   \n",
       "26         A/wood duck/North Carolina/W24-026/2024    H5N1     1.0    H5N1   \n",
       "38                                 A/Texas/37/2024    H5N1     3.0    H5N1   \n",
       "40       A/canada_goose/Indiana/24-005146-001/2024    H5N1     4.0    H5N1   \n",
       "\n",
       "    H5 Clade  ...                Genus            Family Molecule_type Length  \\\n",
       "0        NaN  ...                  NaN               NaN           NaN    NaN   \n",
       "16       NaN  ...  Alphainfluenzavirus  Orthomyxoviridae      ssRNA(-)  252.0   \n",
       "26       NaN  ...  Alphainfluenzavirus  Orthomyxoviridae      ssRNA(-)   90.0   \n",
       "38       NaN  ...  Alphainfluenzavirus  Orthomyxoviridae      ssRNA(-)   90.0   \n",
       "40  2.3.4.4b  ...  Alphainfluenzavirus  Orthomyxoviridae      ssRNA(-)  252.0   \n",
       "\n",
       "   Genotype Segment         Protein  USA        Host Collection_Date  \n",
       "0       NaN     NaN             NaN  NaN         NaN             NaN  \n",
       "16     H5N1       3    PA-X protein   CA  Bos taurus       12/9/2024  \n",
       "26     H5N1       2  PB1-F2 protein   WA  Lynx rufus            2024  \n",
       "38     H5N1       2  PB1-F2 protein   WA  Lynx rufus            2024  \n",
       "40     H5N1       3    PA-X protein   WA  Lynx rufus            2024  \n",
       "\n",
       "[5 rows x 44 columns]"
      ]
     },
     "execution_count": 206,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Drop the duplicates on 'SRA_Accession' column from the concat\n",
    "h5n1_records_df['SRA_Accession'] = h5n1_records_df['SRA_Accession'].astype(str)\n",
    "h5n1_records_df.drop_duplicates(subset=['SRA_Accession'], keep='first', inplace=True)\n",
    "h5n1_records_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Serovar</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>...</th>\n",
       "      <th>Organism_Name</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Nucleotide</th>\n",
       "      <th>Molecule_type</th>\n",
       "      <th>Length</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Protein</th>\n",
       "      <th>USA</th>\n",
       "      <th>Host</th>\n",
       "      <th>Collection_Date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>11320.54589</td>\n",
       "      <td>Influenza A virus A/hooded merganser/South Car...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047523175.1</td>\n",
       "      <td>PV070684</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>252.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PA-X protein</td>\n",
       "      <td>CA</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>12/9/2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>11320.54595</td>\n",
       "      <td>Influenza A virus A/wood duck/North Carolina/W...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047521275.1</td>\n",
       "      <td>PV070691</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>90.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PB1-F2 protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>11320.54640</td>\n",
       "      <td>Influenza A virus A/Texas/37/2024</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047521295.1</td>\n",
       "      <td>PV070699</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>90.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PB1-F2 protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>11320.57078</td>\n",
       "      <td>Influenza A virus A/canada_goose/Indiana/24-00...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>...</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047521295.1</td>\n",
       "      <td>PV070700</td>\n",
       "      <td>ssRNA(-)</td>\n",
       "      <td>252.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PA-X protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 39 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Genome ID                                        Genome Name  \\\n",
       "0   11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "16  11320.54589  Influenza A virus A/hooded merganser/South Car...   \n",
       "26  11320.54595  Influenza A virus A/wood duck/North Carolina/W...   \n",
       "38  11320.54640                  Influenza A virus A/Texas/37/2024   \n",
       "40  11320.57078  Influenza A virus A/canada_goose/Indiana/24-00...   \n",
       "\n",
       "              Family                Genus                         Species  \\\n",
       "0   Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "16  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "26  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "38  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "40  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                            Strain Serovar  Segment Subtype  \\\n",
       "0              A/mute swan/Mangystau/1-S24R-2/2024    H5N1      1.0    H5N1   \n",
       "16  A/hooded merganser/South Carolina/W24-028/2024    H5N1      8.0    H5N1   \n",
       "26         A/wood duck/North Carolina/W24-026/2024    H5N1      1.0    H5N1   \n",
       "38                                 A/Texas/37/2024    H5N1      3.0    H5N1   \n",
       "40       A/canada_goose/Indiana/24-005146-001/2024    H5N1      4.0    H5N1   \n",
       "\n",
       "    H5 Clade  ...      Organism_Name         Assembly Nucleotide  \\\n",
       "0        NaN  ...                NaN              NaN        NaN   \n",
       "16       NaN  ...  Influenza A virus  GCA_047523175.1   PV070684   \n",
       "26       NaN  ...  Influenza A virus  GCA_047521275.1   PV070691   \n",
       "38       NaN  ...  Influenza A virus  GCA_047521295.1   PV070699   \n",
       "40  2.3.4.4b  ...  Influenza A virus  GCA_047521295.1   PV070700   \n",
       "\n",
       "   Molecule_type Length Genotype         Protein  USA        Host  \\\n",
       "0            NaN    NaN      NaN             NaN  NaN         NaN   \n",
       "16      ssRNA(-)  252.0     H5N1    PA-X protein   CA  Bos taurus   \n",
       "26      ssRNA(-)   90.0     H5N1  PB1-F2 protein   WA  Lynx rufus   \n",
       "38      ssRNA(-)   90.0     H5N1  PB1-F2 protein   WA  Lynx rufus   \n",
       "40      ssRNA(-)  252.0     H5N1    PA-X protein   WA  Lynx rufus   \n",
       "\n",
       "   Collection_Date  \n",
       "0              NaN  \n",
       "16       12/9/2024  \n",
       "26            2024  \n",
       "38            2024  \n",
       "40            2024  \n",
       "\n",
       "[5 rows x 39 columns]"
      ]
     },
     "execution_count": 207,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove duplicate column names\n",
    "h5n1_records_df = h5n1_records_df.loc[:, ~h5n1_records_df.columns.duplicated()]\n",
    "h5n1_records_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Genome ID</th>\n",
       "      <th>Genome Name</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Segment</th>\n",
       "      <th>Subtype</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>Subclade</th>\n",
       "      <th>...</th>\n",
       "      <th>Accession</th>\n",
       "      <th>Organism_Name</th>\n",
       "      <th>Assembly</th>\n",
       "      <th>Nucleotide</th>\n",
       "      <th>Length</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Protein</th>\n",
       "      <th>USA</th>\n",
       "      <th>Host</th>\n",
       "      <th>Collection_Date</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>11320.50624</td>\n",
       "      <td>Influenza A virus A/mute swan/Mangystau/1-S24R...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>1.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>...</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>11320.54589</td>\n",
       "      <td>Influenza A virus A/hooded merganser/South Car...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>8.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>...</td>\n",
       "      <td>XOD09042</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047523175.1</td>\n",
       "      <td>PV070684</td>\n",
       "      <td>252.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PA-X protein</td>\n",
       "      <td>CA</td>\n",
       "      <td>Bos taurus</td>\n",
       "      <td>12/9/2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>11320.54595</td>\n",
       "      <td>Influenza A virus A/wood duck/North Carolina/W...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>1.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>...</td>\n",
       "      <td>XOD09052</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047521275.1</td>\n",
       "      <td>PV070691</td>\n",
       "      <td>90.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PB1-F2 protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>11320.54640</td>\n",
       "      <td>Influenza A virus A/Texas/37/2024</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>3.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B3.13</td>\n",
       "      <td>...</td>\n",
       "      <td>XOD09064</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047521295.1</td>\n",
       "      <td>PV070699</td>\n",
       "      <td>90.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PB1-F2 protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>11320.57078</td>\n",
       "      <td>Influenza A virus A/canada_goose/Indiana/24-00...</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>4.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.7</td>\n",
       "      <td>...</td>\n",
       "      <td>XOD09066</td>\n",
       "      <td>Influenza A virus</td>\n",
       "      <td>GCA_047521295.1</td>\n",
       "      <td>PV070700</td>\n",
       "      <td>252.0</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>PA-X protein</td>\n",
       "      <td>WA</td>\n",
       "      <td>Lynx rufus</td>\n",
       "      <td>2024</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>5 rows × 34 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "      Genome ID                                        Genome Name  \\\n",
       "0   11320.50624  Influenza A virus A/mute swan/Mangystau/1-S24R...   \n",
       "16  11320.54589  Influenza A virus A/hooded merganser/South Car...   \n",
       "26  11320.54595  Influenza A virus A/wood duck/North Carolina/W...   \n",
       "38  11320.54640                  Influenza A virus A/Texas/37/2024   \n",
       "40  11320.57078  Influenza A virus A/canada_goose/Indiana/24-00...   \n",
       "\n",
       "              Family                Genus                         Species  \\\n",
       "0   Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "16  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "26  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "38  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "40  Orthomyxoviridae  Alphainfluenzavirus  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                            Strain  Segment Subtype  H5 Clade  \\\n",
       "0              A/mute swan/Mangystau/1-S24R-2/2024      1.0    H5N1       NaN   \n",
       "16  A/hooded merganser/South Carolina/W24-028/2024      8.0    H5N1       NaN   \n",
       "26         A/wood duck/North Carolina/W24-026/2024      1.0    H5N1       NaN   \n",
       "38                                 A/Texas/37/2024      3.0    H5N1       NaN   \n",
       "40       A/canada_goose/Indiana/24-005146-001/2024      4.0    H5N1  2.3.4.4b   \n",
       "\n",
       "   Subclade  ... Accession      Organism_Name         Assembly Nucleotide  \\\n",
       "0       NaN  ...       NaN                NaN              NaN        NaN   \n",
       "16     C2.1  ...  XOD09042  Influenza A virus  GCA_047523175.1   PV070684   \n",
       "26     C2.1  ...  XOD09052  Influenza A virus  GCA_047521275.1   PV070691   \n",
       "38    B3.13  ...  XOD09064  Influenza A virus  GCA_047521295.1   PV070699   \n",
       "40     B3.7  ...  XOD09066  Influenza A virus  GCA_047521295.1   PV070700   \n",
       "\n",
       "   Length  Genotype         Protein  USA        Host Collection_Date  \n",
       "0     NaN       NaN             NaN  NaN         NaN             NaN  \n",
       "16  252.0      H5N1    PA-X protein   CA  Bos taurus       12/9/2024  \n",
       "26   90.0      H5N1  PB1-F2 protein   WA  Lynx rufus            2024  \n",
       "38   90.0      H5N1  PB1-F2 protein   WA  Lynx rufus            2024  \n",
       "40  252.0      H5N1    PA-X protein   WA  Lynx rufus            2024  \n",
       "\n",
       "[5 rows x 34 columns]"
      ]
     },
     "execution_count": 208,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove the column names not specific for this model\n",
    "h5n1_records_df.drop(columns=['Serovar','Geographic Group','Host Name','Lab Host','Molecule_type'],inplace=True) \n",
    "h5n1_records_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 6288 entries, 0 to 72348\n",
      "Data columns (total 13 columns):\n",
      " #   Column               Non-Null Count  Dtype  \n",
      "---  ------               --------------  -----  \n",
      " 0   Nucleotide           6250 non-null   object \n",
      " 1   GC Content           4010 non-null   float64\n",
      " 2   Family               4010 non-null   object \n",
      " 3   Genus                4010 non-null   object \n",
      " 4   Species              4010 non-null   object \n",
      " 5   Strain               4010 non-null   object \n",
      " 6   Genotype             6250 non-null   object \n",
      " 7   Segment              4010 non-null   float64\n",
      " 8   H5 Clade             3312 non-null   object \n",
      " 9   Subclade             3222 non-null   object \n",
      " 10  Host Group           3936 non-null   object \n",
      " 11  Geographic Location  4010 non-null   object \n",
      " 12  Host Common Name     3936 non-null   object \n",
      "dtypes: float64(2), object(11)\n",
      "memory usage: 687.8+ KB\n"
     ]
    }
   ],
   "source": [
    "# Create a dataframe with genetic info\n",
    "h5n1_genetic_df=h5n1_records_df[['Nucleotide','GC Content','Family','Genus','Species','Strain','Genotype','Segment','H5 Clade','Subclade','Host Group','Geographic Location','Host Common Name']]\n",
    "h5n1_genetic_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Nucleotide</th>\n",
       "      <th>GC Content</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Segment</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>Subclade</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Common Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>NaN</td>\n",
       "      <td>44.396920</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/mute swan/Mangystau/1-S24R-2/2024</td>\n",
       "      <td>NaN</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "      <td>Avian</td>\n",
       "      <td>Kazakhstan: Mangystau</td>\n",
       "      <td>Mute Swan</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PV070684</td>\n",
       "      <td>45.254630</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Hooded Merganser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PV070691</td>\n",
       "      <td>45.419190</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Wood duck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PV070699</td>\n",
       "      <td>43.976448</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B3.13</td>\n",
       "      <td>Human</td>\n",
       "      <td>USA: Texas</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PV070700</td>\n",
       "      <td>41.901410</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.7</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Goose</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Nucleotide  GC Content            Family                Genus  \\\n",
       "0        NaN   44.396920  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "1   PV070684   45.254630  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "2   PV070691   45.419190  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "3   PV070699   43.976448  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "4   PV070700   41.901410  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "\n",
       "                          Species  \\\n",
       "0  Alphainfluenzavirus influenzae   \n",
       "1  Alphainfluenzavirus influenzae   \n",
       "2  Alphainfluenzavirus influenzae   \n",
       "3  Alphainfluenzavirus influenzae   \n",
       "4  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                           Strain Genotype  Segment  H5 Clade  \\\n",
       "0             A/mute swan/Mangystau/1-S24R-2/2024      NaN      1.0       NaN   \n",
       "1  A/hooded merganser/South Carolina/W24-028/2024     H5N1      8.0       NaN   \n",
       "2         A/wood duck/North Carolina/W24-026/2024     H5N1      1.0       NaN   \n",
       "3                                 A/Texas/37/2024     H5N1      3.0       NaN   \n",
       "4       A/canada_goose/Indiana/24-005146-001/2024     H5N1      4.0  2.3.4.4b   \n",
       "\n",
       "  Subclade Host Group    Geographic Location  Host Common Name  \n",
       "0      NaN      Avian  Kazakhstan: Mangystau         Mute Swan  \n",
       "1     C2.1      Avian                    USA  Hooded Merganser  \n",
       "2     C2.1      Avian                    USA         Wood duck  \n",
       "3    B3.13      Human             USA: Texas             Human  \n",
       "4     B3.7      Avian                    USA             Goose  "
      ]
     },
     "execution_count": 210,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Remove duplicate values of Nucleotide\n",
    "h5n1_genetic_df=h5n1_genetic_df.drop_duplicates(subset=['Nucleotide'], keep='first').reset_index(drop=True)\n",
    "h5n1_genetic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 6247 entries, 1 to 6247\n",
      "Data columns (total 13 columns):\n",
      " #   Column               Non-Null Count  Dtype  \n",
      "---  ------               --------------  -----  \n",
      " 0   Nucleotide           6247 non-null   object \n",
      " 1   GC Content           3969 non-null   float64\n",
      " 2   Family               3969 non-null   object \n",
      " 3   Genus                3969 non-null   object \n",
      " 4   Species              3969 non-null   object \n",
      " 5   Strain               3969 non-null   object \n",
      " 6   Genotype             6247 non-null   object \n",
      " 7   Segment              3969 non-null   float64\n",
      " 8   H5 Clade             3274 non-null   object \n",
      " 9   Subclade             3215 non-null   object \n",
      " 10  Host Group           3895 non-null   object \n",
      " 11  Geographic Location  3969 non-null   object \n",
      " 12  Host Common Name     3895 non-null   object \n",
      "dtypes: float64(2), object(11)\n",
      "memory usage: 683.3+ KB\n"
     ]
    }
   ],
   "source": [
    "# drop rows with Nucleotide null values\n",
    "h5n1_genetic_df=h5n1_genetic_df.dropna(subset=['Nucleotide'])\n",
    "h5n1_genetic_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Nucleotide</th>\n",
       "      <th>GC Content</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Segment</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>Subclade</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Common Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PV070684</td>\n",
       "      <td>45.254630</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Hooded Merganser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PV070691</td>\n",
       "      <td>45.419190</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Wood duck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PV070699</td>\n",
       "      <td>43.976448</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>NaN</td>\n",
       "      <td>B3.13</td>\n",
       "      <td>Human</td>\n",
       "      <td>USA: Texas</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PV070700</td>\n",
       "      <td>41.901410</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.7</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Goose</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PV070708</td>\n",
       "      <td>45.823387</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/chicken/Ohio/24-005334-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.6</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Chicken</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Nucleotide  GC Content            Family                Genus  \\\n",
       "1   PV070684   45.254630  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "2   PV070691   45.419190  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "3   PV070699   43.976448  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "4   PV070700   41.901410  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "5   PV070708   45.823387  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "\n",
       "                          Species  \\\n",
       "1  Alphainfluenzavirus influenzae   \n",
       "2  Alphainfluenzavirus influenzae   \n",
       "3  Alphainfluenzavirus influenzae   \n",
       "4  Alphainfluenzavirus influenzae   \n",
       "5  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                           Strain Genotype  Segment  H5 Clade  \\\n",
       "1  A/hooded merganser/South Carolina/W24-028/2024     H5N1      8.0       NaN   \n",
       "2         A/wood duck/North Carolina/W24-026/2024     H5N1      1.0       NaN   \n",
       "3                                 A/Texas/37/2024     H5N1      3.0       NaN   \n",
       "4       A/canada_goose/Indiana/24-005146-001/2024     H5N1      4.0  2.3.4.4b   \n",
       "5               A/chicken/Ohio/24-005334-001/2024     H5N1      8.0  2.3.4.4b   \n",
       "\n",
       "  Subclade Host Group Geographic Location  Host Common Name  \n",
       "1     C2.1      Avian                 USA  Hooded Merganser  \n",
       "2     C2.1      Avian                 USA         Wood duck  \n",
       "3    B3.13      Human          USA: Texas             Human  \n",
       "4     B3.7      Avian                 USA             Goose  \n",
       "5     B3.6      Avian                 USA           Chicken  "
      ]
     },
     "execution_count": 212,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "h5n1_genetic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 3215 entries, 1 to 3969\n",
      "Data columns (total 13 columns):\n",
      " #   Column               Non-Null Count  Dtype  \n",
      "---  ------               --------------  -----  \n",
      " 0   Nucleotide           3215 non-null   object \n",
      " 1   GC Content           3215 non-null   float64\n",
      " 2   Family               3215 non-null   object \n",
      " 3   Genus                3215 non-null   object \n",
      " 4   Species              3215 non-null   object \n",
      " 5   Strain               3215 non-null   object \n",
      " 6   Genotype             3215 non-null   object \n",
      " 7   Segment              3215 non-null   float64\n",
      " 8   H5 Clade             2579 non-null   object \n",
      " 9   Subclade             3215 non-null   object \n",
      " 10  Host Group           3160 non-null   object \n",
      " 11  Geographic Location  3215 non-null   object \n",
      " 12  Host Common Name     3160 non-null   object \n",
      "dtypes: float64(2), object(11)\n",
      "memory usage: 351.6+ KB\n"
     ]
    }
   ],
   "source": [
    "# drop rows with subclade null values\n",
    "h5n1_genetic_df=h5n1_genetic_df.dropna(subset=['Subclade'])\n",
    "h5n1_genetic_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 3160 entries, 1 to 3969\n",
      "Data columns (total 13 columns):\n",
      " #   Column               Non-Null Count  Dtype  \n",
      "---  ------               --------------  -----  \n",
      " 0   Nucleotide           3160 non-null   object \n",
      " 1   GC Content           3160 non-null   float64\n",
      " 2   Family               3160 non-null   object \n",
      " 3   Genus                3160 non-null   object \n",
      " 4   Species              3160 non-null   object \n",
      " 5   Strain               3160 non-null   object \n",
      " 6   Genotype             3160 non-null   object \n",
      " 7   Segment              3160 non-null   float64\n",
      " 8   H5 Clade             2546 non-null   object \n",
      " 9   Subclade             3160 non-null   object \n",
      " 10  Host Group           3160 non-null   object \n",
      " 11  Geographic Location  3160 non-null   object \n",
      " 12  Host Common Name     3160 non-null   object \n",
      "dtypes: float64(2), object(11)\n",
      "memory usage: 345.6+ KB\n"
     ]
    }
   ],
   "source": [
    "# Delete rows with missing values for Host Group / Host group name\n",
    "h5n1_genetic_df=h5n1_genetic_df.dropna(subset=['Host Group','Host Common Name'])\n",
    "h5n1_genetic_df.info()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Nucleotide</th>\n",
       "      <th>GC Content</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Segment</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>Subclade</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Common Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PV070684</td>\n",
       "      <td>45.254630</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Hooded Merganser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PV070691</td>\n",
       "      <td>45.419190</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Wood duck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PV070699</td>\n",
       "      <td>43.976448</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.13</td>\n",
       "      <td>Human</td>\n",
       "      <td>USA: Texas</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PV070700</td>\n",
       "      <td>41.901410</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.7</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Goose</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PV070708</td>\n",
       "      <td>45.823387</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/chicken/Ohio/24-005334-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.6</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Chicken</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Nucleotide  GC Content            Family                Genus  \\\n",
       "1   PV070684   45.254630  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "2   PV070691   45.419190  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "3   PV070699   43.976448  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "4   PV070700   41.901410  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "5   PV070708   45.823387  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "\n",
       "                          Species  \\\n",
       "1  Alphainfluenzavirus influenzae   \n",
       "2  Alphainfluenzavirus influenzae   \n",
       "3  Alphainfluenzavirus influenzae   \n",
       "4  Alphainfluenzavirus influenzae   \n",
       "5  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                           Strain Genotype  Segment  H5 Clade  \\\n",
       "1  A/hooded merganser/South Carolina/W24-028/2024     H5N1      8.0  2.3.4.4b   \n",
       "2         A/wood duck/North Carolina/W24-026/2024     H5N1      1.0  2.3.4.4b   \n",
       "3                                 A/Texas/37/2024     H5N1      3.0  2.3.4.4b   \n",
       "4       A/canada_goose/Indiana/24-005146-001/2024     H5N1      4.0  2.3.4.4b   \n",
       "5               A/chicken/Ohio/24-005334-001/2024     H5N1      8.0  2.3.4.4b   \n",
       "\n",
       "  Subclade Host Group Geographic Location  Host Common Name  \n",
       "1     C2.1      Avian                 USA  Hooded Merganser  \n",
       "2     C2.1      Avian                 USA         Wood duck  \n",
       "3    B3.13      Human          USA: Texas             Human  \n",
       "4     B3.7      Avian                 USA             Goose  \n",
       "5     B3.6      Avian                 USA           Chicken  "
      ]
     },
     "execution_count": 215,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "# Fill the missing H5 Clade values to 2.3.4.4b\n",
    "\n",
    "h5n1_genetic_df['H5 Clade']= '2.3.4.4b'\n",
    "h5n1_genetic_df.head()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Nucleotide</th>\n",
       "      <th>GC Content</th>\n",
       "      <th>Family</th>\n",
       "      <th>Genus</th>\n",
       "      <th>Species</th>\n",
       "      <th>Strain</th>\n",
       "      <th>Genotype</th>\n",
       "      <th>Segment</th>\n",
       "      <th>H5 Clade</th>\n",
       "      <th>Subclade</th>\n",
       "      <th>Host Group</th>\n",
       "      <th>Geographic Location</th>\n",
       "      <th>Host Common Name</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>PV070684</td>\n",
       "      <td>45.254630</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/hooded merganser/South Carolina/W24-028/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Hooded Merganser</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>PV070691</td>\n",
       "      <td>45.419190</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/wood duck/North Carolina/W24-026/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>1.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>C2.1</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Wood duck</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>PV070699</td>\n",
       "      <td>43.976448</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/Texas/37/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>3.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.13</td>\n",
       "      <td>Human</td>\n",
       "      <td>USA: Texas</td>\n",
       "      <td>Human</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>PV070700</td>\n",
       "      <td>41.901410</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/canada_goose/Indiana/24-005146-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.7</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Goose</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>PV070708</td>\n",
       "      <td>45.823387</td>\n",
       "      <td>Orthomyxoviridae</td>\n",
       "      <td>Alphainfluenzavirus</td>\n",
       "      <td>Alphainfluenzavirus influenzae</td>\n",
       "      <td>A/chicken/Ohio/24-005334-001/2024</td>\n",
       "      <td>H5N1</td>\n",
       "      <td>8.0</td>\n",
       "      <td>2.3.4.4b</td>\n",
       "      <td>B3.6</td>\n",
       "      <td>Avian</td>\n",
       "      <td>USA</td>\n",
       "      <td>Chicken</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "  Nucleotide  GC Content            Family                Genus  \\\n",
       "1   PV070684   45.254630  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "2   PV070691   45.419190  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "3   PV070699   43.976448  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "4   PV070700   41.901410  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "5   PV070708   45.823387  Orthomyxoviridae  Alphainfluenzavirus   \n",
       "\n",
       "                          Species  \\\n",
       "1  Alphainfluenzavirus influenzae   \n",
       "2  Alphainfluenzavirus influenzae   \n",
       "3  Alphainfluenzavirus influenzae   \n",
       "4  Alphainfluenzavirus influenzae   \n",
       "5  Alphainfluenzavirus influenzae   \n",
       "\n",
       "                                           Strain Genotype  Segment  H5 Clade  \\\n",
       "1  A/hooded merganser/South Carolina/W24-028/2024     H5N1      8.0  2.3.4.4b   \n",
       "2         A/wood duck/North Carolina/W24-026/2024     H5N1      1.0  2.3.4.4b   \n",
       "3                                 A/Texas/37/2024     H5N1      3.0  2.3.4.4b   \n",
       "4       A/canada_goose/Indiana/24-005146-001/2024     H5N1      4.0  2.3.4.4b   \n",
       "5               A/chicken/Ohio/24-005334-001/2024     H5N1      8.0  2.3.4.4b   \n",
       "\n",
       "  Subclade Host Group Geographic Location  Host Common Name  \n",
       "1     C2.1      Avian                 USA  Hooded Merganser  \n",
       "2     C2.1      Avian                 USA         Wood duck  \n",
       "3    B3.13      Human          USA: Texas             Human  \n",
       "4     B3.7      Avian                 USA             Goose  \n",
       "5     B3.6      Avian                 USA           Chicken  "
      ]
     },
     "execution_count": 216,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "#h5n1_genetic_df['Nucleotide'].to_csv(\"nucleotide_number.csv\",index=False)\n",
    "nucleotide_list=h5n1_genetic_df['Nucleotide'].tolist()\n",
    "#print(nucleotide_list)\n",
    "\n",
    "# Write the final dataframe as csv file\n",
    "h5n1_genetic_df.to_csv(\"../Data/h5n1_genetic_data.csv\",index_label=False)\n",
    "h5n1_genetic_df.head()\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "<class 'pandas.core.frame.DataFrame'>\n",
      "Index: 3160 entries, 1 to 3969\n",
      "Data columns (total 11 columns):\n",
      " #   Column               Non-Null Count  Dtype  \n",
      "---  ------               --------------  -----  \n",
      " 0   GC Content           3160 non-null   float64\n",
      " 1   Family               3160 non-null   object \n",
      " 2   Genus                3160 non-null   object \n",
      " 3   Species              3160 non-null   object \n",
      " 4   Genotype             3160 non-null   object \n",
      " 5   Segment              3160 non-null   float64\n",
      " 6   H5 Clade             3160 non-null   object \n",
      " 7   Subclade             3160 non-null   object \n",
      " 8   Host Group           3160 non-null   object \n",
      " 9   Geographic Location  3160 non-null   object \n",
      " 10  Host Common Name     3160 non-null   object \n",
      "dtypes: float64(2), object(9)\n",
      "memory usage: 296.2+ KB\n"
     ]
    }
   ],
   "source": [
    "#Drop Nucleotide column\n",
    "h5n1_genetic_df=h5n1_genetic_df.drop(columns=['Nucleotide','Strain'])\n",
    "h5n1_genetic_df.info()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Data Processing\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "PV070684.1\n",
      "ATGGAAGACTTTGTGCGACAATGCTTCAATCCAATGATTGTCGAGCTTGCGGAAAAAGCAATGAAAGAATATGGGGAAGATCCGAAAATCGAGACAAACAAATTTGCCGCAATATGCACACACTTAGAAGTCTGTTTCATGTATTCGGATTTCCATTTTATTGACGAACGAGGCGAATCAATGATTGTAGAATCTGGCGATCCAAATGCATTATTGAAACACCGATTTGAGATAATCGAAGGGAGAGACCGAGCAATGGCCTGGACAGTGGTGAATAGTATCTGCAACACCACAGGGGTCGAAAAGCCCAAATTCCTCCCTGATTTGTATGACTACAGAGAGAACAGATTCATTGAAATTGGAGTAACGCGAAGGGAAGTTCACATATACTATTTGGAAAAAGCCAACAAGATAAAATCAGAGAAAACACATATTCACATATTCTCATTCACTGGAGAGGAAATGGCCACCAAGGCGGACTACACCCTTGATGAAGAGAGCAGAGCAAGAATAAAAACCAGACTGTTCACTATAAGACAAGAAATGGCCAGTAGAGGTCTATGGGATTCCTTTCGTCAATCCGAGAGAGGCGAAGAGACAATTGAAGAAAGATTTGAAATCACAGGAACCATGCGCAGGCTTGCCGACCAAAGTATTCCACCGAACTTCTCCAGCCTTGAAAACTTTAGAGCCTATGTGGATGGATTCGAACCGAACGGCTGCATTGAGGGCAAGCTTTCTCAAATGTCAAAAGAGGTGAACGCCAGAATTGAGCCATTTCTGAAGACAACACCACGCCCTCTCAGATTACCTGATGGGCCTCCCTGTCCTCAGCGGTCGAAGTTCTTGCTGATGGATGCCCTTAAGTTGAGCATCGAAGACCCTAGTCATGAGGGGGAGGGCATACCGCTGTATGATGCAATCAAATGCATGAAGACATTTTTTGGCTGGAAAGAGCCCAACATCGTAAAGCCGCATGAAAAAGGCATAAACCCTAATTACCTCCTGGCTTGGAAGCAGGTGCTGGCAGAACTTCAAGACATTGAAAATGAGGAGAAAATTCCAAAAACAAAGAACATGAAGAAAACAAGCCAATTGAAGTGGGCACTTGGTGAGAACATGGCTCCAGAAAAAGTGGACTTTGAGGACTGCAAAGATGTTAGCGATCTAAGACAGTACGACAGTGACGAACCAGAGTCTAGATCACTAGCAAGCTGGATTCAGAGTGAATTCAACAAGGCATGCGAACTGACAGATTCGAGTTGGATTGAACTTGATGAGATAGGGGAAGACATTGCTCCAATCGAACACATTGCGAGTGTGAGGAGGAACTATTTCACAGCGGAGGTATCCCATTGCAGGGCCACTGAATACATAATGAAGGGAGTATACATAAACACAGCCCTATTGAATGCATCCTGTGCAGCCATGGATGACTTCCAATTGATTCCAATAATAAGTAAGTGCAGAACTAAAGAAGGAAGACGGAGGACAAATCTGTATGGATTCATTATAAAAGGAAGATCCCATTTGAGGAATGACACCGATGTGGTAAACTTTGTGAGCATGGAATTCTCTCTAACTGACCCGAGGCTAGAGCCACACAAATGGGAAAAGTACTGTGTTCTTGAGATAGGAGACATGCTCCTACGGACTGCAATAGGCCAAGTATTGAGGCCCATGTTCCTGTATGTGAGAACCAATGGAACTTCCAAGATCAAAATGAAATGGGGCATGGAGATGAGGCGATGCCTTCTTCAGTCCCTTCAACAAATTGAGAGCATGATTGAGGCCGAATCTTCTGTCAAAGAGAAGGACATGTCCAAGGAATTCTTTGAAAACAAATCAGAAACATGGCCAATTGGAGAATCACCCAAAGGGGTGGAGGAAGGCTCTATTGGGAAAGTATGCAGAACATTGCTAGCAAAGTCTGTATTCAACAGCCTATATGCATCTCCTCAACTCGAGGGGTTTTCAGCTGAATCAAGAAAATTGCTTCTCATTGTTCAGGCACTTAGGGACAACCTGGAACCTGGAACCTTCGATCTTGGGGGGCTATATGAAGCAATTGAGGAGTGCCTGATTAACGATCCCTGGGTTTTGCTTAATGCATCTTGGTTCAACTCCTTCCTCACACATGCACTGAAATAG\n",
      "2151\n"
     ]
    }
   ],
   "source": [
    "\n",
    "# Read FASTA sequences from file\n",
    "\n",
    "from Bio import Entrez, SeqIO\n",
    "\n",
    "sequences = list(SeqIO.parse(\"sequence.fasta\", \"fasta\"))\n",
    "\n",
    "# Print first sequence\n",
    "print(sequences[0].id)\n",
    "print(sequences[0].seq)\n",
    "print(len(sequences[0].seq))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "defaultdict(<class 'int'>, {Seq('ATGGAAGA'): 1, Seq('TGGAAGAC'): 1, Seq('GGAAGACT'): 1, Seq('GAAGACTT'): 1, Seq('AAGACTTT'): 1, Seq('AGACTTTG'): 1, Seq('GACTTTGT'): 1, Seq('ACTTTGTG'): 2, Seq('CTTTGTGC'): 1, Seq('TTTGTGCG'): 1, Seq('TTGTGCGA'): 1, Seq('TGTGCGAC'): 1, Seq('GTGCGACA'): 1, Seq('TGCGACAA'): 1, Seq('GCGACAAT'): 1, Seq('CGACAATG'): 1, Seq('GACAATGC'): 1, Seq('ACAATGCT'): 1, Seq('CAATGCTT'): 1, Seq('AATGCTTC'): 1, Seq('ATGCTTCA'): 1, Seq('TGCTTCAA'): 1, Seq('GCTTCAAT'): 1, Seq('CTTCAATC'): 1, Seq('TTCAATCC'): 1, Seq('TCAATCCA'): 1, Seq('CAATCCAA'): 1, Seq('AATCCAAT'): 1, Seq('ATCCAATG'): 1, Seq('TCCAATGA'): 1, Seq('CCAATGAT'): 1, Seq('CAATGATT'): 2, Seq('AATGATTG'): 2, Seq('ATGATTGT'): 2, Seq('TGATTGTC'): 1, Seq('GATTGTCG'): 1, Seq('ATTGTCGA'): 1, Seq('TTGTCGAG'): 1, Seq('TGTCGAGC'): 1, Seq('GTCGAGCT'): 1, Seq('TCGAGCTT'): 1, Seq('CGAGCTTG'): 1, Seq('GAGCTTGC'): 1, Seq('AGCTTGCG'): 1, Seq('GCTTGCGG'): 1, Seq('CTTGCGGA'): 1, Seq('TTGCGGAA'): 1, Seq('TGCGGAAA'): 1, Seq('GCGGAAAA'): 1, Seq('CGGAAAAA'): 1, Seq('GGAAAAAG'): 2, Seq('GAAAAAGC'): 2, Seq('AAAAAGCA'): 1, Seq('AAAAGCAA'): 1, Seq('AAAGCAAT'): 1, Seq('AAGCAATG'): 1, Seq('AGCAATGA'): 1, Seq('GCAATGAA'): 1, Seq('CAATGAAA'): 1, Seq('AATGAAAG'): 1, Seq('ATGAAAGA'): 1, Seq('TGAAAGAA'): 1, Seq('GAAAGAAT'): 1, Seq('AAAGAATA'): 1, Seq('AAGAATAT'): 1, Seq('AGAATATG'): 1, Seq('GAATATGG'): 1, Seq('AATATGGG'): 1, Seq('ATATGGGG'): 1, Seq('TATGGGGA'): 1, Seq('ATGGGGAA'): 1, Seq('TGGGGAAG'): 1, Seq('GGGGAAGA'): 2, Seq('GGGAAGAT'): 1, Seq('GGAAGATC'): 2, Seq('GAAGATCC'): 2, Seq('AAGATCCG'): 1, Seq('AGATCCGA'): 1, Seq('GATCCGAA'): 1, Seq('ATCCGAAA'): 1, Seq('TCCGAAAA'): 1, Seq('CCGAAAAT'): 1, Seq('CGAAAATC'): 1, Seq('GAAAATCG'): 1, Seq('AAAATCGA'): 1, Seq('AAATCGAG'): 1, Seq('AATCGAGA'): 1, Seq('ATCGAGAC'): 1, Seq('TCGAGACA'): 1, Seq('CGAGACAA'): 1, Seq('GAGACAAA'): 1, Seq('AGACAAAC'): 1, Seq('GACAAACA'): 1, Seq('ACAAACAA'): 1, Seq('CAAACAAA'): 1, Seq('AAACAAAT'): 2, Seq('AACAAATT'): 2, Seq('ACAAATTT'): 1, Seq('CAAATTTG'): 1, Seq('AAATTTGC'): 1, Seq('AATTTGCC'): 1, Seq('ATTTGCCG'): 1, Seq('TTTGCCGC'): 1, Seq('TTGCCGCA'): 1, Seq('TGCCGCAA'): 1, Seq('GCCGCAAT'): 1, Seq('CCGCAATA'): 1, Seq('CGCAATAT'): 1, Seq('GCAATATG'): 1, Seq('CAATATGC'): 1, Seq('AATATGCA'): 1, Seq('ATATGCAC'): 1, Seq('TATGCACA'): 1, Seq('ATGCACAC'): 1, Seq('TGCACACA'): 1, Seq('GCACACAC'): 1, Seq('CACACACT'): 1, Seq('ACACACTT'): 1, Seq('CACACTTA'): 1, Seq('ACACTTAG'): 1, Seq('CACTTAGA'): 1, Seq('ACTTAGAA'): 1, Seq('CTTAGAAG'): 1, Seq('TTAGAAGT'): 1, Seq('TAGAAGTC'): 1, Seq('AGAAGTCT'): 1, Seq('GAAGTCTG'): 1, Seq('AAGTCTGT'): 2, Seq('AGTCTGTT'): 1, Seq('GTCTGTTT'): 1, Seq('TCTGTTTC'): 1, Seq('CTGTTTCA'): 1, Seq('TGTTTCAT'): 1, Seq('GTTTCATG'): 1, Seq('TTTCATGT'): 1, Seq('TTCATGTA'): 1, Seq('TCATGTAT'): 1, Seq('CATGTATT'): 1, Seq('ATGTATTC'): 1, Seq('TGTATTCG'): 1, Seq('GTATTCGG'): 1, Seq('TATTCGGA'): 1, Seq('ATTCGGAT'): 1, Seq('TTCGGATT'): 1, Seq('TCGGATTT'): 1, Seq('CGGATTTC'): 1, Seq('GGATTTCC'): 1, Seq('GATTTCCA'): 1, Seq('ATTTCCAT'): 1, Seq('TTTCCATT'): 1, Seq('TTCCATTT'): 1, Seq('TCCATTTT'): 1, Seq('CCATTTTA'): 1, Seq('CATTTTAT'): 1, Seq('ATTTTATT'): 1, Seq('TTTTATTG'): 1, Seq('TTTATTGA'): 1, Seq('TTATTGAC'): 1, Seq('TATTGACG'): 1, Seq('ATTGACGA'): 1, Seq('TTGACGAA'): 1, Seq('TGACGAAC'): 2, Seq('GACGAACG'): 1, Seq('ACGAACGA'): 1, Seq('CGAACGAG'): 1, Seq('GAACGAGG'): 1, Seq('AACGAGGC'): 1, Seq('ACGAGGCG'): 1, Seq('CGAGGCGA'): 1, Seq('GAGGCGAA'): 2, Seq('AGGCGAAT'): 1, Seq('GGCGAATC'): 1, Seq('GCGAATCA'): 1, Seq('CGAATCAA'): 1, Seq('GAATCAAT'): 1, Seq('AATCAATG'): 1, Seq('ATCAATGA'): 1, Seq('TCAATGAT'): 1, Seq('TGATTGTA'): 1, Seq('GATTGTAG'): 1, Seq('ATTGTAGA'): 1, Seq('TTGTAGAA'): 1, Seq('TGTAGAAT'): 1, Seq('GTAGAATC'): 1, Seq('TAGAATCT'): 1, Seq('AGAATCTG'): 1, Seq('GAATCTGG'): 1, Seq('AATCTGGC'): 1, Seq('ATCTGGCG'): 1, Seq('TCTGGCGA'): 1, Seq('CTGGCGAT'): 1, Seq('TGGCGATC'): 1, Seq('GGCGATCC'): 1, Seq('GCGATCCA'): 1, Seq('CGATCCAA'): 1, Seq('GATCCAAA'): 1, Seq('ATCCAAAT'): 1, Seq('TCCAAATG'): 1, Seq('CCAAATGC'): 1, Seq('CAAATGCA'): 2, Seq('AAATGCAT'): 2, Seq('AATGCATT'): 1, Seq('ATGCATTA'): 1, Seq('TGCATTAT'): 1, Seq('GCATTATT'): 1, Seq('CATTATTG'): 1, Seq('ATTATTGA'): 1, Seq('TTATTGAA'): 1, Seq('TATTGAAA'): 1, Seq('ATTGAAAC'): 1, Seq('TTGAAACA'): 1, Seq('TGAAACAC'): 1, Seq('GAAACACC'): 1, Seq('AAACACCG'): 1, Seq('AACACCGA'): 1, Seq('ACACCGAT'): 2, Seq('CACCGATT'): 1, Seq('ACCGATTT'): 1, Seq('CCGATTTG'): 1, Seq('CGATTTGA'): 1, Seq('GATTTGAG'): 1, Seq('ATTTGAGA'): 1, Seq('TTTGAGAT'): 1, Seq('TTGAGATA'): 2, Seq('TGAGATAA'): 1, Seq('GAGATAAT'): 1, Seq('AGATAATC'): 1, Seq('GATAATCG'): 1, Seq('ATAATCGA'): 1, Seq('TAATCGAA'): 1, Seq('AATCGAAG'): 1, Seq('ATCGAAGG'): 1, Seq('TCGAAGGG'): 1, Seq('CGAAGGGA'): 2, Seq('GAAGGGAG'): 2, Seq('AAGGGAGA'): 1, Seq('AGGGAGAG'): 1, Seq('GGGAGAGA'): 1, Seq('GGAGAGAC'): 1, Seq('GAGAGACC'): 1, Seq('AGAGACCG'): 1, Seq('GAGACCGA'): 1, Seq('AGACCGAG'): 1, Seq('GACCGAGC'): 1, Seq('ACCGAGCA'): 1, Seq('CCGAGCAA'): 1, Seq('CGAGCAAT'): 1, Seq('GAGCAATG'): 1, Seq('AGCAATGG'): 1, Seq('GCAATGGC'): 1, Seq('CAATGGCC'): 1, Seq('AATGGCCT'): 1, Seq('ATGGCCTG'): 1, Seq('TGGCCTGG'): 1, Seq('GGCCTGGA'): 1, Seq('GCCTGGAC'): 1, Seq('CCTGGACA'): 1, Seq('CTGGACAG'): 1, Seq('TGGACAGT'): 1, Seq('GGACAGTG'): 1, Seq('GACAGTGG'): 1, Seq('ACAGTGGT'): 1, Seq('CAGTGGTG'): 1, Seq('AGTGGTGA'): 1, Seq('GTGGTGAA'): 1, Seq('TGGTGAAT'): 1, Seq('GGTGAATA'): 1, Seq('GTGAATAG'): 1, Seq('TGAATAGT'): 1, Seq('GAATAGTA'): 1, Seq('AATAGTAT'): 1, Seq('ATAGTATC'): 1, Seq('TAGTATCT'): 1, Seq('AGTATCTG'): 1, Seq('GTATCTGC'): 1, Seq('TATCTGCA'): 1, Seq('ATCTGCAA'): 1, Seq('TCTGCAAC'): 1, Seq('CTGCAACA'): 1, Seq('TGCAACAC'): 1, Seq('GCAACACC'): 1, Seq('CAACACCA'): 2, Seq('AACACCAC'): 2, Seq('ACACCACA'): 1, Seq('CACCACAG'): 1, Seq('ACCACAGG'): 1, Seq('CCACAGGG'): 1, Seq('CACAGGGG'): 1, Seq('ACAGGGGT'): 1, Seq('CAGGGGTC'): 1, Seq('AGGGGTCG'): 1, Seq('GGGGTCGA'): 1, Seq('GGGTCGAA'): 1, Seq('GGTCGAAA'): 1, Seq('GTCGAAAA'): 1, Seq('TCGAAAAG'): 1, Seq('CGAAAAGC'): 1, Seq('GAAAAGCC'): 1, Seq('AAAAGCCC'): 1, Seq('AAAGCCCA'): 1, Seq('AAGCCCAA'): 1, Seq('AGCCCAAA'): 1, Seq('GCCCAAAT'): 1, Seq('CCCAAATT'): 1, Seq('CCAAATTC'): 1, Seq('CAAATTCC'): 1, Seq('AAATTCCT'): 1, Seq('AATTCCTC'): 1, Seq('ATTCCTCC'): 1, Seq('TTCCTCCC'): 1, Seq('TCCTCCCT'): 1, Seq('CCTCCCTG'): 2, Seq('CTCCCTGA'): 1, Seq('TCCCTGAT'): 1, Seq('CCCTGATT'): 1, Seq('CCTGATTT'): 1, Seq('CTGATTTG'): 1, Seq('TGATTTGT'): 1, Seq('GATTTGTA'): 1, Seq('ATTTGTAT'): 1, Seq('TTTGTATG'): 1, Seq('TTGTATGA'): 1, Seq('TGTATGAC'): 1, Seq('GTATGACT'): 1, Seq('TATGACTA'): 1, Seq('ATGACTAC'): 1, Seq('TGACTACA'): 1, Seq('GACTACAG'): 1, Seq('ACTACAGA'): 1, Seq('CTACAGAG'): 1, Seq('TACAGAGA'): 1, Seq('ACAGAGAG'): 1, Seq('CAGAGAGA'): 1, Seq('AGAGAGAA'): 1, Seq('GAGAGAAC'): 1, Seq('AGAGAACA'): 1, Seq('GAGAACAG'): 1, Seq('AGAACAGA'): 1, Seq('GAACAGAT'): 1, Seq('AACAGATT'): 1, Seq('ACAGATTC'): 2, Seq('CAGATTCA'): 1, Seq('AGATTCAT'): 1, Seq('GATTCATT'): 2, Seq('ATTCATTG'): 1, Seq('TTCATTGA'): 1, Seq('TCATTGAA'): 1, Seq('CATTGAAA'): 2, Seq('ATTGAAAT'): 1, Seq('TTGAAATT'): 1, Seq('TGAAATTG'): 1, Seq('GAAATTGG'): 1, Seq('AAATTGGA'): 1, Seq('AATTGGAG'): 2, Seq('ATTGGAGT'): 1, Seq('TTGGAGTA'): 1, Seq('TGGAGTAA'): 1, Seq('GGAGTAAC'): 1, Seq('GAGTAACG'): 1, Seq('AGTAACGC'): 1, Seq('GTAACGCG'): 1, Seq('TAACGCGA'): 1, Seq('AACGCGAA'): 1, Seq('ACGCGAAG'): 1, Seq('CGCGAAGG'): 1, Seq('GCGAAGGG'): 1, Seq('GAAGGGAA'): 1, Seq('AAGGGAAG'): 1, Seq('AGGGAAGT'): 1, Seq('GGGAAGTT'): 1, Seq('GGAAGTTC'): 1, Seq('GAAGTTCA'): 1, Seq('AAGTTCAC'): 1, Seq('AGTTCACA'): 1, Seq('GTTCACAT'): 1, Seq('TTCACATA'): 2, Seq('TCACATAT'): 2, Seq('CACATATA'): 1, Seq('ACATATAC'): 1, Seq('CATATACT'): 1, Seq('ATATACTA'): 1, Seq('TATACTAT'): 1, Seq('ATACTATT'): 1, Seq('TACTATTT'): 1, Seq('ACTATTTG'): 1, Seq('CTATTTGG'): 1, Seq('TATTTGGA'): 1, Seq('ATTTGGAA'): 1, Seq('TTTGGAAA'): 1, Seq('TTGGAAAA'): 1, Seq('TGGAAAAA'): 1, Seq('AAAAAGCC'): 1, Seq('AAAAGCCA'): 1, Seq('AAAGCCAA'): 1, Seq('AAGCCAAC'): 1, Seq('AGCCAACA'): 1, Seq('GCCAACAA'): 1, Seq('CCAACAAG'): 1, Seq('CAACAAGA'): 1, Seq('AACAAGAT'): 1, Seq('ACAAGATA'): 1, Seq('CAAGATAA'): 1, Seq('AAGATAAA'): 1, Seq('AGATAAAA'): 1, Seq('GATAAAAT'): 1, Seq('ATAAAATC'): 1, Seq('TAAAATCA'): 1, Seq('AAAATCAG'): 1, Seq('AAATCAGA'): 2, Seq('AATCAGAG'): 1, Seq('ATCAGAGA'): 1, Seq('TCAGAGAA'): 1, Seq('CAGAGAAA'): 1, Seq('AGAGAAAA'): 1, Seq('GAGAAAAC'): 1, Seq('AGAAAACA'): 2, Seq('GAAAACAC'): 1, Seq('AAAACACA'): 1, Seq('AAACACAT'): 1, Seq('AACACATA'): 1, Seq('ACACATAT'): 1, Seq('CACATATT'): 2, Seq('ACATATTC'): 2, Seq('CATATTCA'): 1, Seq('ATATTCAC'): 1, Seq('TATTCACA'): 1, Seq('ATTCACAT'): 1, Seq('CATATTCT'): 1, Seq('ATATTCTC'): 1, Seq('TATTCTCA'): 1, Seq('ATTCTCAT'): 1, Seq('TTCTCATT'): 2, Seq('TCTCATTC'): 1, Seq('CTCATTCA'): 1, Seq('TCATTCAC'): 1, Seq('CATTCACT'): 1, Seq('ATTCACTG'): 1, Seq('TTCACTGG'): 1, Seq('TCACTGGA'): 1, Seq('CACTGGAG'): 1, Seq('ACTGGAGA'): 1, Seq('CTGGAGAG'): 1, Seq('TGGAGAGG'): 1, Seq('GGAGAGGA'): 1, Seq('GAGAGGAA'): 1, Seq('AGAGGAAA'): 1, Seq('GAGGAAAT'): 1, Seq('AGGAAATG'): 1, Seq('GGAAATGG'): 1, Seq('GAAATGGC'): 2, Seq('AAATGGCC'): 2, Seq('AATGGCCA'): 2, Seq('ATGGCCAC'): 1, Seq('TGGCCACC'): 1, Seq('GGCCACCA'): 1, Seq('GCCACCAA'): 1, Seq('CCACCAAG'): 1, Seq('CACCAAGG'): 1, Seq('ACCAAGGC'): 1, Seq('CCAAGGCG'): 1, Seq('CAAGGCGG'): 1, Seq('AAGGCGGA'): 1, Seq('AGGCGGAC'): 1, Seq('GGCGGACT'): 1, Seq('GCGGACTA'): 1, Seq('CGGACTAC'): 1, Seq('GGACTACA'): 1, Seq('GACTACAC'): 1, Seq('ACTACACC'): 1, Seq('CTACACCC'): 1, Seq('TACACCCT'): 1, Seq('ACACCCTT'): 1, Seq('CACCCTTG'): 1, Seq('ACCCTTGA'): 1, Seq('CCCTTGAT'): 1, Seq('CCTTGATG'): 1, Seq('CTTGATGA'): 2, Seq('TTGATGAA'): 1, Seq('TGATGAAG'): 1, Seq('GATGAAGA'): 1, Seq('ATGAAGAG'): 1, Seq('TGAAGAGA'): 1, Seq('GAAGAGAG'): 1, Seq('AAGAGAGC'): 1, Seq('AGAGAGCA'): 1, Seq('GAGAGCAG'): 1, Seq('AGAGCAGA'): 1, Seq('GAGCAGAG'): 1, Seq('AGCAGAGC'): 1, Seq('GCAGAGCA'): 1, Seq('CAGAGCAA'): 1, Seq('AGAGCAAG'): 1, Seq('GAGCAAGA'): 1, Seq('AGCAAGAA'): 1, Seq('GCAAGAAT'): 1, Seq('CAAGAATA'): 1, Seq('AAGAATAA'): 1, Seq('AGAATAAA'): 1, Seq('GAATAAAA'): 1, Seq('AATAAAAA'): 1, Seq('ATAAAAAC'): 1, Seq('TAAAAACC'): 1, Seq('AAAAACCA'): 1, Seq('AAAACCAG'): 1, Seq('AAACCAGA'): 1, Seq('AACCAGAC'): 1, Seq('ACCAGACT'): 1, Seq('CCAGACTG'): 1, Seq('CAGACTGT'): 1, Seq('AGACTGTT'): 1, Seq('GACTGTTC'): 1, Seq('ACTGTTCA'): 1, Seq('CTGTTCAC'): 1, Seq('TGTTCACT'): 1, Seq('GTTCACTA'): 1, Seq('TTCACTAT'): 1, Seq('TCACTATA'): 1, Seq('CACTATAA'): 1, Seq('ACTATAAG'): 1, Seq('CTATAAGA'): 1, Seq('TATAAGAC'): 1, Seq('ATAAGACA'): 1, Seq('TAAGACAA'): 1, Seq('AAGACAAG'): 1, Seq('AGACAAGA'): 1, Seq('GACAAGAA'): 1, Seq('ACAAGAAA'): 1, Seq('CAAGAAAT'): 1, Seq('AAGAAATG'): 1, Seq('AGAAATGG'): 1, Seq('ATGGCCAG'): 1, Seq('TGGCCAGT'): 1, Seq('GGCCAGTA'): 1, Seq('GCCAGTAG'): 1, Seq('CCAGTAGA'): 1, Seq('CAGTAGAG'): 1, Seq('AGTAGAGG'): 1, Seq('GTAGAGGT'): 1, Seq('TAGAGGTC'): 1, Seq('AGAGGTCT'): 1, Seq('GAGGTCTA'): 1, Seq('AGGTCTAT'): 1, Seq('GGTCTATG'): 1, Seq('GTCTATGG'): 1, Seq('TCTATGGG'): 1, Seq('CTATGGGA'): 1, Seq('TATGGGAT'): 1, Seq('ATGGGATT'): 1, Seq('TGGGATTC'): 1, Seq('GGGATTCC'): 1, Seq('GGATTCCT'): 1, Seq('GATTCCTT'): 1, Seq('ATTCCTTT'): 1, Seq('TTCCTTTC'): 1, Seq('TCCTTTCG'): 1, Seq('CCTTTCGT'): 1, Seq('CTTTCGTC'): 1, Seq('TTTCGTCA'): 1, Seq('TTCGTCAA'): 1, Seq('TCGTCAAT'): 1, Seq('CGTCAATC'): 1, Seq('GTCAATCC'): 1, Seq('TCAATCCG'): 1, Seq('CAATCCGA'): 1, Seq('AATCCGAG'): 1, Seq('ATCCGAGA'): 1, Seq('TCCGAGAG'): 1, Seq('CCGAGAGA'): 1, Seq('CGAGAGAG'): 1, Seq('GAGAGAGG'): 1, Seq('AGAGAGGC'): 1, Seq('GAGAGGCG'): 1, Seq('AGAGGCGA'): 1, Seq('AGGCGAAG'): 1, Seq('GGCGAAGA'): 1, Seq('GCGAAGAG'): 1, Seq('CGAAGAGA'): 1, Seq('GAAGAGAC'): 1, Seq('AAGAGACA'): 1, Seq('AGAGACAA'): 1, Seq('GAGACAAT'): 1, Seq('AGACAATT'): 1, Seq('GACAATTG'): 1, Seq('ACAATTGA'): 1, Seq('CAATTGAA'): 2, Seq('AATTGAAG'): 2, Seq('ATTGAAGA'): 1, Seq('TTGAAGAA'): 1, Seq('TGAAGAAA'): 2, Seq('GAAGAAAG'): 1, Seq('AAGAAAGA'): 1, Seq('AGAAAGAT'): 1, Seq('GAAAGATT'): 1, Seq('AAAGATTT'): 1, Seq('AAGATTTG'): 1, Seq('AGATTTGA'): 1, Seq('GATTTGAA'): 1, Seq('ATTTGAAA'): 1, Seq('TTTGAAAT'): 1, Seq('TTGAAATC'): 1, Seq('TGAAATCA'): 1, Seq('GAAATCAC'): 1, Seq('AAATCACA'): 1, Seq('AATCACAG'): 1, Seq('ATCACAGG'): 1, Seq('TCACAGGA'): 1, Seq('CACAGGAA'): 1, Seq('ACAGGAAC'): 1, Seq('CAGGAACC'): 1, Seq('AGGAACCA'): 1, Seq('GGAACCAT'): 1, Seq('GAACCATG'): 1, Seq('AACCATGC'): 1, Seq('ACCATGCG'): 1, Seq('CCATGCGC'): 1, Seq('CATGCGCA'): 1, Seq('ATGCGCAG'): 1, Seq('TGCGCAGG'): 1, Seq('GCGCAGGC'): 1, Seq('CGCAGGCT'): 1, Seq('GCAGGCTT'): 1, Seq('CAGGCTTG'): 1, Seq('AGGCTTGC'): 1, Seq('GGCTTGCC'): 1, Seq('GCTTGCCG'): 1, Seq('CTTGCCGA'): 1, Seq('TTGCCGAC'): 1, Seq('TGCCGACC'): 1, Seq('GCCGACCA'): 1, Seq('CCGACCAA'): 1, Seq('CGACCAAA'): 1, Seq('GACCAAAG'): 1, Seq('ACCAAAGT'): 1, Seq('CCAAAGTA'): 1, Seq('CAAAGTAT'): 1, Seq('AAAGTATT'): 1, Seq('AAGTATTC'): 1, Seq('AGTATTCC'): 1, Seq('GTATTCCA'): 1, Seq('TATTCCAC'): 1, Seq('ATTCCACC'): 1, Seq('TTCCACCG'): 1, Seq('TCCACCGA'): 1, Seq('CCACCGAA'): 1, Seq('CACCGAAC'): 1, Seq('ACCGAACT'): 1, Seq('CCGAACTT'): 1, Seq('CGAACTTC'): 1, Seq('GAACTTCT'): 1, Seq('AACTTCTC'): 1, Seq('ACTTCTCC'): 1, Seq('CTTCTCCA'): 1, Seq('TTCTCCAG'): 1, Seq('TCTCCAGC'): 1, Seq('CTCCAGCC'): 1, Seq('TCCAGCCT'): 1, Seq('CCAGCCTT'): 1, Seq('CAGCCTTG'): 1, Seq('AGCCTTGA'): 1, Seq('GCCTTGAA'): 1, Seq('CCTTGAAA'): 1, Seq('CTTGAAAA'): 1, Seq('TTGAAAAC'): 2, Seq('TGAAAACT'): 1, Seq('GAAAACTT'): 1, Seq('AAAACTTT'): 1, Seq('AAACTTTA'): 1, Seq('AACTTTAG'): 1, Seq('ACTTTAGA'): 1, Seq('CTTTAGAG'): 1, Seq('TTTAGAGC'): 1, Seq('TTAGAGCC'): 1, Seq('TAGAGCCT'): 1, Seq('AGAGCCTA'): 1, Seq('GAGCCTAT'): 1, Seq('AGCCTATG'): 1, Seq('GCCTATGT'): 1, Seq('CCTATGTG'): 1, Seq('CTATGTGG'): 1, Seq('TATGTGGA'): 1, Seq('ATGTGGAT'): 1, Seq('TGTGGATG'): 1, Seq('GTGGATGG'): 1, Seq('TGGATGGA'): 1, Seq('GGATGGAT'): 1, Seq('GATGGATT'): 1, Seq('ATGGATTC'): 2, Seq('TGGATTCG'): 1, Seq('GGATTCGA'): 1, Seq('GATTCGAA'): 1, Seq('ATTCGAAC'): 1, Seq('TTCGAACC'): 1, Seq('TCGAACCG'): 1, Seq('CGAACCGA'): 1, Seq('GAACCGAA'): 1, Seq('AACCGAAC'): 1, Seq('ACCGAACG'): 1, Seq('CCGAACGG'): 1, Seq('CGAACGGC'): 1, Seq('GAACGGCT'): 1, Seq('AACGGCTG'): 1, Seq('ACGGCTGC'): 1, Seq('CGGCTGCA'): 1, Seq('GGCTGCAT'): 1, Seq('GCTGCATT'): 1, Seq('CTGCATTG'): 1, Seq('TGCATTGA'): 1, Seq('GCATTGAG'): 1, Seq('CATTGAGG'): 1, Seq('ATTGAGGG'): 1, Seq('TTGAGGGC'): 1, Seq('TGAGGGCA'): 1, Seq('GAGGGCAA'): 1, Seq('AGGGCAAG'): 1, Seq('GGGCAAGC'): 1, Seq('GGCAAGCT'): 1, Seq('GCAAGCTT'): 1, Seq('CAAGCTTT'): 1, Seq('AAGCTTTC'): 1, Seq('AGCTTTCT'): 1, Seq('GCTTTCTC'): 1, Seq('CTTTCTCA'): 1, Seq('TTTCTCAA'): 1, Seq('TTCTCAAA'): 1, Seq('TCTCAAAT'): 1, Seq('CTCAAATG'): 1, Seq('TCAAATGT'): 1, Seq('CAAATGTC'): 1, Seq('AAATGTCA'): 1, Seq('AATGTCAA'): 1, Seq('ATGTCAAA'): 1, Seq('TGTCAAAA'): 1, Seq('GTCAAAAG'): 1, Seq('TCAAAAGA'): 1, Seq('CAAAAGAG'): 1, Seq('AAAAGAGG'): 1, Seq('AAAGAGGT'): 1, Seq('AAGAGGTG'): 1, Seq('AGAGGTGA'): 1, Seq('GAGGTGAA'): 1, Seq('AGGTGAAC'): 1, Seq('GGTGAACG'): 1, Seq('GTGAACGC'): 1, Seq('TGAACGCC'): 1, Seq('GAACGCCA'): 1, Seq('AACGCCAG'): 1, Seq('ACGCCAGA'): 1, Seq('CGCCAGAA'): 1, Seq('GCCAGAAT'): 1, Seq('CCAGAATT'): 1, Seq('CAGAATTG'): 1, Seq('AGAATTGA'): 1, Seq('GAATTGAG'): 1, Seq('AATTGAGC'): 1, Seq('ATTGAGCC'): 1, Seq('TTGAGCCA'): 1, Seq('TGAGCCAT'): 1, Seq('GAGCCATT'): 1, Seq('AGCCATTT'): 1, Seq('GCCATTTC'): 1, Seq('CCATTTCT'): 1, Seq('CATTTCTG'): 1, Seq('ATTTCTGA'): 1, Seq('TTTCTGAA'): 1, Seq('TTCTGAAG'): 1, Seq('TCTGAAGA'): 1, Seq('CTGAAGAC'): 1, Seq('TGAAGACA'): 2, Seq('GAAGACAA'): 1, Seq('AAGACAAC'): 1, Seq('AGACAACA'): 1, Seq('GACAACAC'): 1, Seq('ACAACACC'): 1, Seq('ACACCACG'): 1, Seq('CACCACGC'): 1, Seq('ACCACGCC'): 1, Seq('CCACGCCC'): 1, Seq('CACGCCCT'): 1, Seq('ACGCCCTC'): 1, Seq('CGCCCTCT'): 1, Seq('GCCCTCTC'): 1, Seq('CCCTCTCA'): 1, Seq('CCTCTCAG'): 1, Seq('CTCTCAGA'): 1, Seq('TCTCAGAT'): 1, Seq('CTCAGATT'): 1, Seq('TCAGATTA'): 1, Seq('CAGATTAC'): 1, Seq('AGATTACC'): 1, Seq('GATTACCT'): 1, Seq('ATTACCTG'): 1, Seq('TTACCTGA'): 1, Seq('TACCTGAT'): 1, Seq('ACCTGATG'): 1, Seq('CCTGATGG'): 1, Seq('CTGATGGG'): 1, Seq('TGATGGGC'): 1, Seq('GATGGGCC'): 1, Seq('ATGGGCCT'): 1, Seq('TGGGCCTC'): 1, Seq('GGGCCTCC'): 1, Seq('GGCCTCCC'): 1, Seq('GCCTCCCT'): 1, Seq('CTCCCTGT'): 1, Seq('TCCCTGTC'): 1, Seq('CCCTGTCC'): 1, Seq('CCTGTCCT'): 1, Seq('CTGTCCTC'): 1, Seq('TGTCCTCA'): 1, Seq('GTCCTCAG'): 1, Seq('TCCTCAGC'): 1, Seq('CCTCAGCG'): 1, Seq('CTCAGCGG'): 1, Seq('TCAGCGGT'): 1, Seq('CAGCGGTC'): 1, Seq('AGCGGTCG'): 1, Seq('GCGGTCGA'): 1, Seq('CGGTCGAA'): 1, Seq('GGTCGAAG'): 1, Seq('GTCGAAGT'): 1, Seq('TCGAAGTT'): 1, Seq('CGAAGTTC'): 1, Seq('GAAGTTCT'): 1, Seq('AAGTTCTT'): 1, Seq('AGTTCTTG'): 1, Seq('GTTCTTGC'): 1, Seq('TTCTTGCT'): 1, Seq('TCTTGCTG'): 1, Seq('CTTGCTGA'): 1, Seq('TTGCTGAT'): 1, Seq('TGCTGATG'): 1, Seq('GCTGATGG'): 1, Seq('CTGATGGA'): 1, Seq('TGATGGAT'): 1, Seq('GATGGATG'): 1, Seq('ATGGATGC'): 1, Seq('TGGATGCC'): 1, Seq('GGATGCCC'): 1, Seq('GATGCCCT'): 1, Seq('ATGCCCTT'): 1, Seq('TGCCCTTA'): 1, Seq('GCCCTTAA'): 1, Seq('CCCTTAAG'): 1, Seq('CCTTAAGT'): 1, Seq('CTTAAGTT'): 1, Seq('TTAAGTTG'): 1, Seq('TAAGTTGA'): 1, Seq('AAGTTGAG'): 1, Seq('AGTTGAGC'): 1, Seq('GTTGAGCA'): 1, Seq('TTGAGCAT'): 1, Seq('TGAGCATC'): 1, Seq('GAGCATCG'): 1, Seq('AGCATCGA'): 1, Seq('GCATCGAA'): 1, Seq('CATCGAAG'): 1, Seq('ATCGAAGA'): 1, Seq('TCGAAGAC'): 1, Seq('CGAAGACC'): 1, Seq('GAAGACCC'): 1, Seq('AAGACCCT'): 1, Seq('AGACCCTA'): 1, Seq('GACCCTAG'): 1, Seq('ACCCTAGT'): 1, Seq('CCCTAGTC'): 1, Seq('CCTAGTCA'): 1, Seq('CTAGTCAT'): 1, Seq('TAGTCATG'): 1, Seq('AGTCATGA'): 1, Seq('GTCATGAG'): 1, Seq('TCATGAGG'): 1, Seq('CATGAGGG'): 1, Seq('ATGAGGGG'): 1, Seq('TGAGGGGG'): 1, Seq('GAGGGGGA'): 1, Seq('AGGGGGAG'): 1, Seq('GGGGGAGG'): 1, Seq('GGGGAGGG'): 1, Seq('GGGAGGGC'): 1, Seq('GGAGGGCA'): 1, Seq('GAGGGCAT'): 1, Seq('AGGGCATA'): 1, Seq('GGGCATAC'): 1, Seq('GGCATACC'): 1, Seq('GCATACCG'): 1, Seq('CATACCGC'): 1, Seq('ATACCGCT'): 1, Seq('TACCGCTG'): 1, Seq('ACCGCTGT'): 1, Seq('CCGCTGTA'): 1, Seq('CGCTGTAT'): 1, Seq('GCTGTATG'): 1, Seq('CTGTATGA'): 1, Seq('TGTATGAT'): 1, Seq('GTATGATG'): 1, Seq('TATGATGC'): 1, Seq('ATGATGCA'): 1, Seq('TGATGCAA'): 1, Seq('GATGCAAT'): 1, Seq('ATGCAATC'): 1, Seq('TGCAATCA'): 1, Seq('GCAATCAA'): 1, Seq('CAATCAAA'): 1, Seq('AATCAAAT'): 1, Seq('ATCAAATG'): 1, Seq('TCAAATGC'): 1, Seq('AATGCATG'): 1, Seq('ATGCATGA'): 1, Seq('TGCATGAA'): 1, Seq('GCATGAAG'): 1, Seq('CATGAAGA'): 2, Seq('ATGAAGAC'): 1, Seq('GAAGACAT'): 2, Seq('AAGACATT'): 3, Seq('AGACATTT'): 1, Seq('GACATTTT'): 1, Seq('ACATTTTT'): 1, Seq('CATTTTTT'): 1, Seq('ATTTTTTG'): 1, Seq('TTTTTTGG'): 1, Seq('TTTTTGGC'): 1, Seq('TTTTGGCT'): 1, Seq('TTTGGCTG'): 1, Seq('TTGGCTGG'): 1, Seq('TGGCTGGA'): 1, Seq('GGCTGGAA'): 1, Seq('GCTGGAAA'): 1, Seq('CTGGAAAG'): 1, Seq('TGGAAAGA'): 1, Seq('GGAAAGAG'): 1, Seq('GAAAGAGC'): 1, Seq('AAAGAGCC'): 1, Seq('AAGAGCCC'): 1, Seq('AGAGCCCA'): 1, Seq('GAGCCCAA'): 1, Seq('AGCCCAAC'): 1, Seq('GCCCAACA'): 1, Seq('CCCAACAT'): 1, Seq('CCAACATC'): 1, Seq('CAACATCG'): 1, Seq('AACATCGT'): 1, Seq('ACATCGTA'): 1, Seq('CATCGTAA'): 1, Seq('ATCGTAAA'): 1, Seq('TCGTAAAG'): 1, Seq('CGTAAAGC'): 1, Seq('GTAAAGCC'): 1, Seq('TAAAGCCG'): 1, Seq('AAAGCCGC'): 1, Seq('AAGCCGCA'): 1, Seq('AGCCGCAT'): 1, Seq('GCCGCATG'): 1, Seq('CCGCATGA'): 1, Seq('CGCATGAA'): 1, Seq('GCATGAAA'): 1, Seq('CATGAAAA'): 1, Seq('ATGAAAAA'): 1, Seq('TGAAAAAG'): 1, Seq('GAAAAAGG'): 1, Seq('AAAAAGGC'): 1, Seq('AAAAGGCA'): 1, Seq('AAAGGCAT'): 1, Seq('AAGGCATA'): 1, Seq('AGGCATAA'): 1, Seq('GGCATAAA'): 1, Seq('GCATAAAC'): 1, Seq('CATAAACC'): 1, Seq('ATAAACCC'): 1, Seq('TAAACCCT'): 1, Seq('AAACCCTA'): 1, Seq('AACCCTAA'): 1, Seq('ACCCTAAT'): 1, Seq('CCCTAATT'): 1, Seq('CCTAATTA'): 1, Seq('CTAATTAC'): 1, Seq('TAATTACC'): 1, Seq('AATTACCT'): 1, Seq('ATTACCTC'): 1, Seq('TTACCTCC'): 1, Seq('TACCTCCT'): 1, Seq('ACCTCCTG'): 1, Seq('CCTCCTGG'): 1, Seq('CTCCTGGC'): 1, Seq('TCCTGGCT'): 1, Seq('CCTGGCTT'): 1, Seq('CTGGCTTG'): 1, Seq('TGGCTTGG'): 1, Seq('GGCTTGGA'): 1, Seq('GCTTGGAA'): 1, Seq('CTTGGAAG'): 1, Seq('TTGGAAGC'): 1, Seq('TGGAAGCA'): 1, Seq('GGAAGCAG'): 1, Seq('GAAGCAGG'): 1, Seq('AAGCAGGT'): 1, Seq('AGCAGGTG'): 1, Seq('GCAGGTGC'): 1, Seq('CAGGTGCT'): 1, Seq('AGGTGCTG'): 1, Seq('GGTGCTGG'): 1, Seq('GTGCTGGC'): 1, Seq('TGCTGGCA'): 1, Seq('GCTGGCAG'): 1, Seq('CTGGCAGA'): 1, Seq('TGGCAGAA'): 1, Seq('GGCAGAAC'): 1, Seq('GCAGAACT'): 2, Seq('CAGAACTT'): 1, Seq('AGAACTTC'): 1, Seq('GAACTTCA'): 1, Seq('AACTTCAA'): 1, Seq('ACTTCAAG'): 1, Seq('CTTCAAGA'): 1, Seq('TTCAAGAC'): 1, Seq('TCAAGACA'): 1, Seq('CAAGACAT'): 1, Seq('AGACATTG'): 2, Seq('GACATTGA'): 1, Seq('ACATTGAA'): 1, Seq('ATTGAAAA'): 1, Seq('TTGAAAAT'): 1, Seq('TGAAAATG'): 1, Seq('GAAAATGA'): 1, Seq('AAAATGAG'): 1, Seq('AAATGAGG'): 1, Seq('AATGAGGA'): 1, Seq('ATGAGGAG'): 1, Seq('TGAGGAGA'): 1, Seq('GAGGAGAA'): 1, Seq('AGGAGAAA'): 1, Seq('GGAGAAAA'): 1, Seq('GAGAAAAT'): 1, Seq('AGAAAATT'): 2, Seq('GAAAATTC'): 1, Seq('AAAATTCC'): 1, Seq('AAATTCCA'): 1, Seq('AATTCCAA'): 1, Seq('ATTCCAAA'): 1, Seq('TTCCAAAA'): 1, Seq('TCCAAAAA'): 1, Seq('CCAAAAAC'): 1, Seq('CAAAAACA'): 1, Seq('AAAAACAA'): 1, Seq('AAAACAAA'): 2, Seq('AAACAAAG'): 1, Seq('AACAAAGA'): 1, Seq('ACAAAGAA'): 1, Seq('CAAAGAAC'): 1, Seq('AAAGAACA'): 1, Seq('AAGAACAT'): 1, Seq('AGAACATG'): 2, Seq('GAACATGA'): 1, Seq('AACATGAA'): 1, Seq('ACATGAAG'): 1, Seq('ATGAAGAA'): 1, Seq('GAAGAAAA'): 1, Seq('AAGAAAAC'): 1, Seq('GAAAACAA'): 2, Seq('AAAACAAG'): 1, Seq('AAACAAGC'): 1, Seq('AACAAGCC'): 1, Seq('ACAAGCCA'): 1, Seq('CAAGCCAA'): 1, Seq('AAGCCAAT'): 1, Seq('AGCCAATT'): 1, Seq('GCCAATTG'): 2, Seq('CCAATTGA'): 2, Seq('ATTGAAGT'): 1, Seq('TTGAAGTG'): 1, Seq('TGAAGTGG'): 1, Seq('GAAGTGGG'): 1, Seq('AAGTGGGC'): 1, Seq('AGTGGGCA'): 1, Seq('GTGGGCAC'): 1, Seq('TGGGCACT'): 1, Seq('GGGCACTT'): 1, Seq('GGCACTTG'): 1, Seq('GCACTTGG'): 1, Seq('CACTTGGT'): 1, Seq('ACTTGGTG'): 1, Seq('CTTGGTGA'): 1, Seq('TTGGTGAG'): 1, Seq('TGGTGAGA'): 1, Seq('GGTGAGAA'): 1, Seq('GTGAGAAC'): 2, Seq('TGAGAACA'): 1, Seq('GAGAACAT'): 1, Seq('GAACATGG'): 1, Seq('AACATGGC'): 2, Seq('ACATGGCT'): 1, Seq('CATGGCTC'): 1, Seq('ATGGCTCC'): 1, Seq('TGGCTCCA'): 1, Seq('GGCTCCAG'): 1, Seq('GCTCCAGA'): 1, Seq('CTCCAGAA'): 1, Seq('TCCAGAAA'): 1, Seq('CCAGAAAA'): 1, Seq('CAGAAAAA'): 1, Seq('AGAAAAAG'): 1, Seq('GAAAAAGT'): 1, Seq('AAAAAGTG'): 1, Seq('AAAAGTGG'): 1, Seq('AAAGTGGA'): 1, Seq('AAGTGGAC'): 1, Seq('AGTGGACT'): 1, Seq('GTGGACTT'): 1, Seq('TGGACTTT'): 1, Seq('GGACTTTG'): 1, Seq('GACTTTGA'): 1, Seq('ACTTTGAG'): 1, Seq('CTTTGAGG'): 1, Seq('TTTGAGGA'): 2, Seq('TTGAGGAC'): 1, Seq('TGAGGACT'): 1, Seq('GAGGACTG'): 1, Seq('AGGACTGC'): 1, Seq('GGACTGCA'): 2, Seq('GACTGCAA'): 2, Seq('ACTGCAAA'): 1, Seq('CTGCAAAG'): 1, Seq('TGCAAAGA'): 1, Seq('GCAAAGAT'): 1, Seq('CAAAGATG'): 1, Seq('AAAGATGT'): 1, Seq('AAGATGTT'): 1, Seq('AGATGTTA'): 1, Seq('GATGTTAG'): 1, Seq('ATGTTAGC'): 1, Seq('TGTTAGCG'): 1, Seq('GTTAGCGA'): 1, Seq('TTAGCGAT'): 1, Seq('TAGCGATC'): 1, Seq('AGCGATCT'): 1, Seq('GCGATCTA'): 1, Seq('CGATCTAA'): 1, Seq('GATCTAAG'): 1, Seq('ATCTAAGA'): 1, Seq('TCTAAGAC'): 1, Seq('CTAAGACA'): 1, Seq('TAAGACAG'): 1, Seq('AAGACAGT'): 1, Seq('AGACAGTA'): 1, Seq('GACAGTAC'): 1, Seq('ACAGTACG'): 1, Seq('CAGTACGA'): 1, Seq('AGTACGAC'): 1, Seq('GTACGACA'): 1, Seq('TACGACAG'): 1, Seq('ACGACAGT'): 1, Seq('CGACAGTG'): 1, Seq('GACAGTGA'): 1, Seq('ACAGTGAC'): 1, Seq('CAGTGACG'): 1, Seq('AGTGACGA'): 1, Seq('GTGACGAA'): 1, Seq('GACGAACC'): 1, Seq('ACGAACCA'): 1, Seq('CGAACCAG'): 1, Seq('GAACCAGA'): 1, Seq('AACCAGAG'): 1, Seq('ACCAGAGT'): 1, Seq('CCAGAGTC'): 1, Seq('CAGAGTCT'): 1, Seq('AGAGTCTA'): 1, Seq('GAGTCTAG'): 1, Seq('AGTCTAGA'): 1, Seq('GTCTAGAT'): 1, Seq('TCTAGATC'): 1, Seq('CTAGATCA'): 1, Seq('TAGATCAC'): 1, Seq('AGATCACT'): 1, Seq('GATCACTA'): 1, Seq('ATCACTAG'): 1, Seq('TCACTAGC'): 1, Seq('CACTAGCA'): 1, Seq('ACTAGCAA'): 1, Seq('CTAGCAAG'): 1, Seq('TAGCAAGC'): 1, Seq('AGCAAGCT'): 1, Seq('GCAAGCTG'): 1, Seq('CAAGCTGG'): 1, Seq('AAGCTGGA'): 1, Seq('AGCTGGAT'): 1, Seq('GCTGGATT'): 1, Seq('CTGGATTC'): 1, Seq('TGGATTCA'): 2, Seq('GGATTCAG'): 1, Seq('GATTCAGA'): 1, Seq('ATTCAGAG'): 1, Seq('TTCAGAGT'): 1, Seq('TCAGAGTG'): 1, Seq('CAGAGTGA'): 1, Seq('AGAGTGAA'): 1, Seq('GAGTGAAT'): 1, Seq('AGTGAATT'): 1, Seq('GTGAATTC'): 1, Seq('TGAATTCA'): 1, Seq('GAATTCAA'): 1, Seq('AATTCAAC'): 1, Seq('ATTCAACA'): 2, Seq('TTCAACAA'): 2, Seq('TCAACAAG'): 1, Seq('CAACAAGG'): 1, Seq('AACAAGGC'): 1, Seq('ACAAGGCA'): 1, Seq('CAAGGCAT'): 1, Seq('AAGGCATG'): 1, Seq('AGGCATGC'): 1, Seq('GGCATGCG'): 1, Seq('GCATGCGA'): 1, Seq('CATGCGAA'): 1, Seq('ATGCGAAC'): 1, Seq('TGCGAACT'): 1, Seq('GCGAACTG'): 1, Seq('CGAACTGA'): 1, Seq('GAACTGAC'): 1, Seq('AACTGACA'): 1, Seq('ACTGACAG'): 1, Seq('CTGACAGA'): 1, Seq('TGACAGAT'): 1, Seq('GACAGATT'): 1, Seq('CAGATTCG'): 1, Seq('AGATTCGA'): 1, Seq('GATTCGAG'): 1, Seq('ATTCGAGT'): 1, Seq('TTCGAGTT'): 1, Seq('TCGAGTTG'): 1, Seq('CGAGTTGG'): 1, Seq('GAGTTGGA'): 1, Seq('AGTTGGAT'): 1, Seq('GTTGGATT'): 1, Seq('TTGGATTG'): 1, Seq('TGGATTGA'): 1, Seq('GGATTGAA'): 1, Seq('GATTGAAC'): 1, Seq('ATTGAACT'): 1, Seq('TTGAACTT'): 1, Seq('TGAACTTG'): 1, Seq('GAACTTGA'): 1, Seq('AACTTGAT'): 1, Seq('ACTTGATG'): 1, Seq('TTGATGAG'): 1, Seq('TGATGAGA'): 1, Seq('GATGAGAT'): 1, Seq('ATGAGATA'): 1, Seq('TGAGATAG'): 2, Seq('GAGATAGG'): 2, Seq('AGATAGGG'): 1, Seq('GATAGGGG'): 1, Seq('ATAGGGGA'): 1, Seq('TAGGGGAA'): 1, Seq('AGGGGAAG'): 1, Seq('GGGAAGAC'): 1, Seq('GGAAGACA'): 1, Seq('GACATTGC'): 1, Seq('ACATTGCT'): 2, Seq('CATTGCTC'): 1, Seq('ATTGCTCC'): 1, Seq('TTGCTCCA'): 1, Seq('TGCTCCAA'): 1, Seq('GCTCCAAT'): 1, Seq('CTCCAATC'): 1, Seq('TCCAATCG'): 1, Seq('CCAATCGA'): 1, Seq('CAATCGAA'): 1, Seq('AATCGAAC'): 1, Seq('ATCGAACA'): 1, Seq('TCGAACAC'): 1, Seq('CGAACACA'): 1, Seq('GAACACAT'): 1, Seq('AACACATT'): 1, Seq('ACACATTG'): 1, Seq('CACATTGC'): 1, Seq('ACATTGCG'): 1, Seq('CATTGCGA'): 1, Seq('ATTGCGAG'): 1, Seq('TTGCGAGT'): 1, Seq('TGCGAGTG'): 1, Seq('GCGAGTGT'): 1, Seq('CGAGTGTG'): 1, Seq('GAGTGTGA'): 1, Seq('AGTGTGAG'): 1, Seq('GTGTGAGG'): 1, Seq('TGTGAGGA'): 1, Seq('GTGAGGAG'): 1, Seq('TGAGGAGG'): 1, Seq('GAGGAGGA'): 1, Seq('AGGAGGAA'): 1, Seq('GGAGGAAC'): 1, Seq('GAGGAACT'): 1, Seq('AGGAACTA'): 1, Seq('GGAACTAT'): 1, Seq('GAACTATT'): 1, Seq('AACTATTT'): 1, Seq('ACTATTTC'): 1, Seq('CTATTTCA'): 1, Seq('TATTTCAC'): 1, Seq('ATTTCACA'): 1, Seq('TTTCACAG'): 1, Seq('TTCACAGC'): 1, Seq('TCACAGCG'): 1, Seq('CACAGCGG'): 1, Seq('ACAGCGGA'): 1, Seq('CAGCGGAG'): 1, Seq('AGCGGAGG'): 1, Seq('GCGGAGGT'): 1, Seq('CGGAGGTA'): 1, Seq('GGAGGTAT'): 1, Seq('GAGGTATC'): 1, Seq('AGGTATCC'): 1, Seq('GGTATCCC'): 1, Seq('GTATCCCA'): 1, Seq('TATCCCAT'): 1, Seq('ATCCCATT'): 2, Seq('TCCCATTG'): 1, Seq('CCCATTGC'): 1, Seq('CCATTGCA'): 1, Seq('CATTGCAG'): 1, Seq('ATTGCAGG'): 1, Seq('TTGCAGGG'): 1, Seq('TGCAGGGC'): 1, Seq('GCAGGGCC'): 1, Seq('CAGGGCCA'): 1, Seq('AGGGCCAC'): 1, Seq('GGGCCACT'): 1, Seq('GGCCACTG'): 1, Seq('GCCACTGA'): 1, Seq('CCACTGAA'): 1, Seq('CACTGAAT'): 1, Seq('ACTGAATA'): 1, Seq('CTGAATAC'): 1, Seq('TGAATACA'): 1, Seq('GAATACAT'): 1, Seq('AATACATA'): 1, Seq('ATACATAA'): 2, Seq('TACATAAT'): 1, Seq('ACATAATG'): 1, Seq('CATAATGA'): 1, Seq('ATAATGAA'): 1, Seq('TAATGAAG'): 1, Seq('AATGAAGG'): 1, Seq('ATGAAGGG'): 1, Seq('TGAAGGGA'): 1, Seq('AAGGGAGT'): 1, Seq('AGGGAGTA'): 1, Seq('GGGAGTAT'): 1, Seq('GGAGTATA'): 1, Seq('GAGTATAC'): 1, Seq('AGTATACA'): 1, Seq('GTATACAT'): 1, Seq('TATACATA'): 1, Seq('TACATAAA'): 1, Seq('ACATAAAC'): 1, Seq('CATAAACA'): 1, Seq('ATAAACAC'): 1, Seq('TAAACACA'): 1, Seq('AAACACAG'): 1, Seq('AACACAGC'): 1, Seq('ACACAGCC'): 1, Seq('CACAGCCC'): 1, Seq('ACAGCCCT'): 1, Seq('CAGCCCTA'): 1, Seq('AGCCCTAT'): 1, Seq('GCCCTATT'): 1, Seq('CCCTATTG'): 1, Seq('CCTATTGA'): 1, Seq('CTATTGAA'): 1, Seq('TATTGAAT'): 1, Seq('ATTGAATG'): 1, Seq('TTGAATGC'): 1, Seq('TGAATGCA'): 1, Seq('GAATGCAT'): 1, Seq('AATGCATC'): 2, Seq('ATGCATCC'): 1, Seq('TGCATCCT'): 1, Seq('GCATCCTG'): 1, Seq('CATCCTGT'): 1, Seq('ATCCTGTG'): 1, Seq('TCCTGTGC'): 1, Seq('CCTGTGCA'): 1, Seq('CTGTGCAG'): 1, Seq('TGTGCAGC'): 1, Seq('GTGCAGCC'): 1, Seq('TGCAGCCA'): 1, Seq('GCAGCCAT'): 1, Seq('CAGCCATG'): 1, Seq('AGCCATGG'): 1, Seq('GCCATGGA'): 1, Seq('CCATGGAT'): 1, Seq('CATGGATG'): 1, Seq('ATGGATGA'): 1, Seq('TGGATGAC'): 1, Seq('GGATGACT'): 1, Seq('GATGACTT'): 1, Seq('ATGACTTC'): 1, Seq('TGACTTCC'): 1, Seq('GACTTCCA'): 1, Seq('ACTTCCAA'): 2, Seq('CTTCCAAT'): 1, Seq('TTCCAATT'): 1, Seq('TCCAATTG'): 1, Seq('CAATTGAT'): 1, Seq('AATTGATT'): 1, Seq('ATTGATTC'): 1, Seq('TTGATTCC'): 1, Seq('TGATTCCA'): 1, Seq('GATTCCAA'): 1, Seq('ATTCCAAT'): 1, Seq('TTCCAATA'): 1, Seq('TCCAATAA'): 1, Seq('CCAATAAT'): 1, Seq('CAATAATA'): 1, Seq('AATAATAA'): 1, Seq('ATAATAAG'): 1, Seq('TAATAAGT'): 1, Seq('AATAAGTA'): 1, Seq('ATAAGTAA'): 1, Seq('TAAGTAAG'): 1, Seq('AAGTAAGT'): 1, Seq('AGTAAGTG'): 1, Seq('GTAAGTGC'): 1, Seq('TAAGTGCA'): 1, Seq('AAGTGCAG'): 1, Seq('AGTGCAGA'): 1, Seq('GTGCAGAA'): 1, Seq('TGCAGAAC'): 2, Seq('CAGAACTA'): 1, Seq('AGAACTAA'): 1, Seq('GAACTAAA'): 1, Seq('AACTAAAG'): 1, Seq('ACTAAAGA'): 1, Seq('CTAAAGAA'): 1, Seq('TAAAGAAG'): 1, Seq('AAAGAAGG'): 1, Seq('AAGAAGGA'): 1, Seq('AGAAGGAA'): 1, Seq('GAAGGAAG'): 1, Seq('AAGGAAGA'): 2, Seq('AGGAAGAC'): 1, Seq('GGAAGACG'): 1, Seq('GAAGACGG'): 1, Seq('AAGACGGA'): 1, Seq('AGACGGAG'): 1, Seq('GACGGAGG'): 1, Seq('ACGGAGGA'): 1, Seq('CGGAGGAC'): 1, Seq('GGAGGACA'): 1, Seq('GAGGACAA'): 1, Seq('AGGACAAA'): 1, Seq('GGACAAAT'): 1, Seq('GACAAATC'): 1, Seq('ACAAATCT'): 1, Seq('CAAATCTG'): 1, Seq('AAATCTGT'): 1, Seq('AATCTGTA'): 1, Seq('ATCTGTAT'): 1, Seq('TCTGTATG'): 1, Seq('CTGTATGG'): 1, Seq('TGTATGGA'): 1, Seq('GTATGGAT'): 1, Seq('TATGGATT'): 1, Seq('GGATTCAT'): 1, Seq('ATTCATTA'): 1, Seq('TTCATTAT'): 1, Seq('TCATTATA'): 1, Seq('CATTATAA'): 1, Seq('ATTATAAA'): 1, Seq('TTATAAAA'): 1, Seq('TATAAAAG'): 1, Seq('ATAAAAGG'): 1, Seq('TAAAAGGA'): 1, Seq('AAAAGGAA'): 1, Seq('AAAGGAAG'): 1, Seq('AGGAAGAT'): 1, Seq('AAGATCCC'): 1, Seq('AGATCCCA'): 1, Seq('GATCCCAT'): 1, Seq('TCCCATTT'): 1, Seq('CCCATTTG'): 1, Seq('CCATTTGA'): 1, Seq('CATTTGAG'): 1, Seq('ATTTGAGG'): 1, Seq('TTGAGGAA'): 1, Seq('TGAGGAAT'): 1, Seq('GAGGAATG'): 1, Seq('AGGAATGA'): 1, Seq('GGAATGAC'): 1, Seq('GAATGACA'): 1, Seq('AATGACAC'): 1, Seq('ATGACACC'): 1, Seq('TGACACCG'): 1, Seq('GACACCGA'): 1, Seq('CACCGATG'): 1, Seq('ACCGATGT'): 1, Seq('CCGATGTG'): 1, Seq('CGATGTGG'): 1, Seq('GATGTGGT'): 1, Seq('ATGTGGTA'): 1, Seq('TGTGGTAA'): 1, Seq('GTGGTAAA'): 1, Seq('TGGTAAAC'): 1, Seq('GGTAAACT'): 1, Seq('GTAAACTT'): 1, Seq('TAAACTTT'): 1, Seq('AAACTTTG'): 1, Seq('AACTTTGT'): 1, Seq('CTTTGTGA'): 1, Seq('TTTGTGAG'): 1, Seq('TTGTGAGC'): 1, Seq('TGTGAGCA'): 1, Seq('GTGAGCAT'): 1, Seq('TGAGCATG'): 1, Seq('GAGCATGG'): 1, Seq('AGCATGGA'): 1, Seq('GCATGGAA'): 1, Seq('CATGGAAT'): 1, Seq('ATGGAATT'): 1, Seq('TGGAATTC'): 1, Seq('GGAATTCT'): 2, Seq('GAATTCTC'): 1, Seq('AATTCTCT'): 1, Seq('ATTCTCTC'): 1, Seq('TTCTCTCT'): 1, Seq('TCTCTCTA'): 1, Seq('CTCTCTAA'): 1, Seq('TCTCTAAC'): 1, Seq('CTCTAACT'): 1, Seq('TCTAACTG'): 1, Seq('CTAACTGA'): 1, Seq('TAACTGAC'): 1, Seq('AACTGACC'): 1, Seq('ACTGACCC'): 1, Seq('CTGACCCG'): 1, Seq('TGACCCGA'): 1, Seq('GACCCGAG'): 1, Seq('ACCCGAGG'): 1, Seq('CCCGAGGC'): 1, Seq('CCGAGGCT'): 1, Seq('CGAGGCTA'): 1, Seq('GAGGCTAG'): 1, Seq('AGGCTAGA'): 1, Seq('GGCTAGAG'): 1, Seq('GCTAGAGC'): 1, Seq('CTAGAGCC'): 1, Seq('TAGAGCCA'): 1, Seq('AGAGCCAC'): 1, Seq('GAGCCACA'): 1, Seq('AGCCACAC'): 1, Seq('GCCACACA'): 1, Seq('CCACACAA'): 1, Seq('CACACAAA'): 1, Seq('ACACAAAT'): 1, Seq('CACAAATG'): 1, Seq('ACAAATGG'): 1, Seq('CAAATGGG'): 1, Seq('AAATGGGA'): 1, Seq('AATGGGAA'): 1, Seq('ATGGGAAA'): 1, Seq('TGGGAAAA'): 1, Seq('GGGAAAAG'): 1, Seq('GGAAAAGT'): 1, Seq('GAAAAGTA'): 1, Seq('AAAAGTAC'): 1, Seq('AAAGTACT'): 1, Seq('AAGTACTG'): 1, Seq('AGTACTGT'): 1, Seq('GTACTGTG'): 1, Seq('TACTGTGT'): 1, Seq('ACTGTGTT'): 1, Seq('CTGTGTTC'): 1, Seq('TGTGTTCT'): 1, Seq('GTGTTCTT'): 1, Seq('TGTTCTTG'): 1, Seq('GTTCTTGA'): 1, Seq('TTCTTGAG'): 1, Seq('TCTTGAGA'): 1, Seq('CTTGAGAT'): 1, Seq('AGATAGGA'): 1, Seq('GATAGGAG'): 1, Seq('ATAGGAGA'): 1, Seq('TAGGAGAC'): 1, Seq('AGGAGACA'): 1, Seq('GGAGACAT'): 1, Seq('GAGACATG'): 1, Seq('AGACATGC'): 1, Seq('GACATGCT'): 1, Seq('ACATGCTC'): 1, Seq('CATGCTCC'): 1, Seq('ATGCTCCT'): 1, Seq('TGCTCCTA'): 1, Seq('GCTCCTAC'): 1, Seq('CTCCTACG'): 1, Seq('TCCTACGG'): 1, Seq('CCTACGGA'): 1, Seq('CTACGGAC'): 1, Seq('TACGGACT'): 1, Seq('ACGGACTG'): 1, Seq('CGGACTGC'): 1, Seq('ACTGCAAT'): 1, Seq('CTGCAATA'): 1, Seq('TGCAATAG'): 1, Seq('GCAATAGG'): 1, Seq('CAATAGGC'): 1, Seq('AATAGGCC'): 1, Seq('ATAGGCCA'): 1, Seq('TAGGCCAA'): 1, Seq('AGGCCAAG'): 1, Seq('GGCCAAGT'): 1, Seq('GCCAAGTA'): 1, Seq('CCAAGTAT'): 1, Seq('CAAGTATT'): 1, Seq('AAGTATTG'): 1, Seq('AGTATTGA'): 1, Seq('GTATTGAG'): 1, Seq('TATTGAGG'): 1, Seq('ATTGAGGC'): 2, Seq('TTGAGGCC'): 2, Seq('TGAGGCCC'): 1, Seq('GAGGCCCA'): 1, Seq('AGGCCCAT'): 1, Seq('GGCCCATG'): 1, Seq('GCCCATGT'): 1, Seq('CCCATGTT'): 1, Seq('CCATGTTC'): 1, Seq('CATGTTCC'): 1, Seq('ATGTTCCT'): 1, Seq('TGTTCCTG'): 1, Seq('GTTCCTGT'): 1, Seq('TTCCTGTA'): 1, Seq('TCCTGTAT'): 1, Seq('CCTGTATG'): 1, Seq('CTGTATGT'): 1, Seq('TGTATGTG'): 1, Seq('GTATGTGA'): 1, Seq('TATGTGAG'): 1, Seq('ATGTGAGA'): 1, Seq('TGTGAGAA'): 1, Seq('TGAGAACC'): 1, Seq('GAGAACCA'): 1, Seq('AGAACCAA'): 1, Seq('GAACCAAT'): 1, Seq('AACCAATG'): 1, Seq('ACCAATGG'): 1, Seq('CCAATGGA'): 1, Seq('CAATGGAA'): 1, Seq('AATGGAAC'): 1, Seq('ATGGAACT'): 1, Seq('TGGAACTT'): 1, Seq('GGAACTTC'): 1, Seq('GAACTTCC'): 1, Seq('AACTTCCA'): 1, Seq('CTTCCAAG'): 1, Seq('TTCCAAGA'): 1, Seq('TCCAAGAT'): 1, Seq('CCAAGATC'): 1, Seq('CAAGATCA'): 1, Seq('AAGATCAA'): 1, Seq('AGATCAAA'): 1, Seq('GATCAAAA'): 1, Seq('ATCAAAAT'): 1, Seq('TCAAAATG'): 1, Seq('CAAAATGA'): 1, Seq('AAAATGAA'): 1, Seq('AAATGAAA'): 1, Seq('AATGAAAT'): 1, Seq('ATGAAATG'): 1, Seq('TGAAATGG'): 1, Seq('GAAATGGG'): 1, Seq('AAATGGGG'): 1, Seq('AATGGGGC'): 1, Seq('ATGGGGCA'): 1, Seq('TGGGGCAT'): 1, Seq('GGGGCATG'): 1, Seq('GGGCATGG'): 1, Seq('GGCATGGA'): 1, Seq('GCATGGAG'): 1, Seq('CATGGAGA'): 1, Seq('ATGGAGAT'): 1, Seq('TGGAGATG'): 1, Seq('GGAGATGA'): 1, Seq('GAGATGAG'): 1, Seq('AGATGAGG'): 1, Seq('GATGAGGC'): 1, Seq('ATGAGGCG'): 1, Seq('TGAGGCGA'): 1, Seq('GAGGCGAT'): 1, Seq('AGGCGATG'): 1, Seq('GGCGATGC'): 1, Seq('GCGATGCC'): 1, Seq('CGATGCCT'): 1, Seq('GATGCCTT'): 1, Seq('ATGCCTTC'): 1, Seq('TGCCTTCT'): 1, Seq('GCCTTCTT'): 1, Seq('CCTTCTTC'): 1, Seq('CTTCTTCA'): 1, Seq('TTCTTCAG'): 1, Seq('TCTTCAGT'): 1, Seq('CTTCAGTC'): 1, Seq('TTCAGTCC'): 1, Seq('TCAGTCCC'): 1, Seq('CAGTCCCT'): 1, Seq('AGTCCCTT'): 1, Seq('GTCCCTTC'): 1, Seq('TCCCTTCA'): 1, Seq('CCCTTCAA'): 1, Seq('CCTTCAAC'): 1, Seq('CTTCAACA'): 1, Seq('TCAACAAA'): 1, Seq('CAACAAAT'): 1, Seq('ACAAATTG'): 1, Seq('CAAATTGA'): 1, Seq('AAATTGAG'): 1, Seq('AATTGAGA'): 1, Seq('ATTGAGAG'): 1, Seq('TTGAGAGC'): 1, Seq('TGAGAGCA'): 1, Seq('GAGAGCAT'): 1, Seq('AGAGCATG'): 1, Seq('GAGCATGA'): 1, Seq('AGCATGAT'): 1, Seq('GCATGATT'): 1, Seq('CATGATTG'): 1, Seq('ATGATTGA'): 1, Seq('TGATTGAG'): 1, Seq('GATTGAGG'): 1, Seq('TGAGGCCG'): 1, Seq('GAGGCCGA'): 1, Seq('AGGCCGAA'): 1, Seq('GGCCGAAT'): 1, Seq('GCCGAATC'): 1, Seq('CCGAATCT'): 1, Seq('CGAATCTT'): 1, Seq('GAATCTTC'): 1, Seq('AATCTTCT'): 1, Seq('ATCTTCTG'): 1, Seq('TCTTCTGT'): 1, Seq('CTTCTGTC'): 1, Seq('TTCTGTCA'): 1, Seq('TCTGTCAA'): 1, Seq('CTGTCAAA'): 1, Seq('TGTCAAAG'): 1, Seq('GTCAAAGA'): 1, Seq('TCAAAGAG'): 1, Seq('CAAAGAGA'): 1, Seq('AAAGAGAA'): 1, Seq('AAGAGAAG'): 1, Seq('AGAGAAGG'): 1, Seq('GAGAAGGA'): 1, Seq('AGAAGGAC'): 1, Seq('GAAGGACA'): 1, Seq('AAGGACAT'): 1, Seq('AGGACATG'): 1, Seq('GGACATGT'): 1, Seq('GACATGTC'): 1, Seq('ACATGTCC'): 1, Seq('CATGTCCA'): 1, Seq('ATGTCCAA'): 1, Seq('TGTCCAAG'): 1, Seq('GTCCAAGG'): 1, Seq('TCCAAGGA'): 1, Seq('CCAAGGAA'): 1, Seq('CAAGGAAT'): 1, Seq('AAGGAATT'): 1, Seq('AGGAATTC'): 1, Seq('GAATTCTT'): 1, Seq('AATTCTTT'): 1, Seq('ATTCTTTG'): 1, Seq('TTCTTTGA'): 1, Seq('TCTTTGAA'): 1, Seq('CTTTGAAA'): 1, Seq('TTTGAAAA'): 1, Seq('TGAAAACA'): 1, Seq('AACAAATC'): 1, Seq('ACAAATCA'): 1, Seq('CAAATCAG'): 1, Seq('AATCAGAA'): 1, Seq('ATCAGAAA'): 1, Seq('TCAGAAAC'): 1, Seq('CAGAAACA'): 1, Seq('AGAAACAT'): 1, Seq('GAAACATG'): 1, Seq('AAACATGG'): 1, Seq('ACATGGCC'): 1, Seq('CATGGCCA'): 1, Seq('ATGGCCAA'): 1, Seq('TGGCCAAT'): 1, Seq('GGCCAATT'): 1, Seq('CCAATTGG'): 1, Seq('CAATTGGA'): 1, Seq('ATTGGAGA'): 1, Seq('TTGGAGAA'): 1, Seq('TGGAGAAT'): 1, Seq('GGAGAATC'): 1, Seq('GAGAATCA'): 1, Seq('AGAATCAC'): 1, Seq('GAATCACC'): 1, Seq('AATCACCC'): 1, Seq('ATCACCCA'): 1, Seq('TCACCCAA'): 1, Seq('CACCCAAA'): 1, Seq('ACCCAAAG'): 1, Seq('CCCAAAGG'): 1, Seq('CCAAAGGG'): 1, Seq('CAAAGGGG'): 1, Seq('AAAGGGGT'): 1, Seq('AAGGGGTG'): 1, Seq('AGGGGTGG'): 1, Seq('GGGGTGGA'): 1, Seq('GGGTGGAG'): 1, Seq('GGTGGAGG'): 1, Seq('GTGGAGGA'): 1, Seq('TGGAGGAA'): 1, Seq('GGAGGAAG'): 1, Seq('GAGGAAGG'): 1, Seq('AGGAAGGC'): 1, Seq('GGAAGGCT'): 1, Seq('GAAGGCTC'): 1, Seq('AAGGCTCT'): 1, Seq('AGGCTCTA'): 1, Seq('GGCTCTAT'): 1, Seq('GCTCTATT'): 1, Seq('CTCTATTG'): 1, Seq('TCTATTGG'): 1, Seq('CTATTGGG'): 1, Seq('TATTGGGA'): 1, Seq('ATTGGGAA'): 1, Seq('TTGGGAAA'): 1, Seq('TGGGAAAG'): 1, Seq('GGGAAAGT'): 1, Seq('GGAAAGTA'): 1, Seq('GAAAGTAT'): 1, Seq('AAAGTATG'): 1, Seq('AAGTATGC'): 1, Seq('AGTATGCA'): 1, Seq('GTATGCAG'): 1, Seq('TATGCAGA'): 1, Seq('ATGCAGAA'): 1, Seq('GCAGAACA'): 1, Seq('CAGAACAT'): 1, Seq('AGAACATT'): 1, Seq('GAACATTG'): 1, Seq('AACATTGC'): 1, Seq('CATTGCTA'): 1, Seq('ATTGCTAG'): 1, Seq('TTGCTAGC'): 1, Seq('TGCTAGCA'): 1, Seq('GCTAGCAA'): 1, Seq('CTAGCAAA'): 1, Seq('TAGCAAAG'): 1, Seq('AGCAAAGT'): 1, Seq('GCAAAGTC'): 1, Seq('CAAAGTCT'): 1, Seq('AAAGTCTG'): 1, Seq('AGTCTGTA'): 1, Seq('GTCTGTAT'): 1, Seq('TCTGTATT'): 1, Seq('CTGTATTC'): 1, Seq('TGTATTCA'): 1, Seq('GTATTCAA'): 1, Seq('TATTCAAC'): 1, Seq('TTCAACAG'): 1, Seq('TCAACAGC'): 1, Seq('CAACAGCC'): 1, Seq('AACAGCCT'): 1, Seq('ACAGCCTA'): 1, Seq('CAGCCTAT'): 1, Seq('AGCCTATA'): 1, Seq('GCCTATAT'): 1, Seq('CCTATATG'): 1, Seq('CTATATGC'): 1, Seq('TATATGCA'): 1, Seq('ATATGCAT'): 1, Seq('TATGCATC'): 1, Seq('ATGCATCT'): 2, Seq('TGCATCTC'): 1, Seq('GCATCTCC'): 1, Seq('CATCTCCT'): 1, Seq('ATCTCCTC'): 1, Seq('TCTCCTCA'): 1, Seq('CTCCTCAA'): 1, Seq('TCCTCAAC'): 1, Seq('CCTCAACT'): 1, Seq('CTCAACTC'): 1, Seq('TCAACTCG'): 1, Seq('CAACTCGA'): 1, Seq('AACTCGAG'): 1, Seq('ACTCGAGG'): 1, Seq('CTCGAGGG'): 1, Seq('TCGAGGGG'): 1, Seq('CGAGGGGT'): 1, Seq('GAGGGGTT'): 1, Seq('AGGGGTTT'): 1, Seq('GGGGTTTT'): 1, Seq('GGGTTTTC'): 1, Seq('GGTTTTCA'): 1, Seq('GTTTTCAG'): 1, Seq('TTTTCAGC'): 1, Seq('TTTCAGCT'): 1, Seq('TTCAGCTG'): 1, Seq('TCAGCTGA'): 1, Seq('CAGCTGAA'): 1, Seq('AGCTGAAT'): 1, Seq('GCTGAATC'): 1, Seq('CTGAATCA'): 1, Seq('TGAATCAA'): 1, Seq('GAATCAAG'): 1, Seq('AATCAAGA'): 1, Seq('ATCAAGAA'): 1, Seq('TCAAGAAA'): 1, Seq('CAAGAAAA'): 1, Seq('AAGAAAAT'): 1, Seq('GAAAATTG'): 1, Seq('AAAATTGC'): 1, Seq('AAATTGCT'): 1, Seq('AATTGCTT'): 1, Seq('ATTGCTTC'): 1, Seq('TTGCTTCT'): 1, Seq('TGCTTCTC'): 1, Seq('GCTTCTCA'): 1, Seq('CTTCTCAT'): 1, Seq('TCTCATTG'): 1, Seq('CTCATTGT'): 1, Seq('TCATTGTT'): 1, Seq('CATTGTTC'): 1, Seq('ATTGTTCA'): 1, Seq('TTGTTCAG'): 1, Seq('TGTTCAGG'): 1, Seq('GTTCAGGC'): 1, Seq('TTCAGGCA'): 1, Seq('TCAGGCAC'): 1, Seq('CAGGCACT'): 1, Seq('AGGCACTT'): 1, Seq('GGCACTTA'): 1, Seq('GCACTTAG'): 1, Seq('CACTTAGG'): 1, Seq('ACTTAGGG'): 1, Seq('CTTAGGGA'): 1, Seq('TTAGGGAC'): 1, Seq('TAGGGACA'): 1, Seq('AGGGACAA'): 1, Seq('GGGACAAC'): 1, Seq('GGACAACC'): 1, Seq('GACAACCT'): 1, Seq('ACAACCTG'): 1, Seq('CAACCTGG'): 1, Seq('AACCTGGA'): 2, Seq('ACCTGGAA'): 2, Seq('CCTGGAAC'): 2, Seq('CTGGAACC'): 2, Seq('TGGAACCT'): 2, Seq('GGAACCTG'): 1, Seq('GAACCTGG'): 1, Seq('GGAACCTT'): 1, Seq('GAACCTTC'): 1, Seq('AACCTTCG'): 1, Seq('ACCTTCGA'): 1, Seq('CCTTCGAT'): 1, Seq('CTTCGATC'): 1, Seq('TTCGATCT'): 1, Seq('TCGATCTT'): 1, Seq('CGATCTTG'): 1, Seq('GATCTTGG'): 1, Seq('ATCTTGGG'): 1, Seq('TCTTGGGG'): 1, Seq('CTTGGGGG'): 1, Seq('TTGGGGGG'): 1, Seq('TGGGGGGC'): 1, Seq('GGGGGGCT'): 1, Seq('GGGGGCTA'): 1, Seq('GGGGCTAT'): 1, Seq('GGGCTATA'): 1, Seq('GGCTATAT'): 1, Seq('GCTATATG'): 1, Seq('CTATATGA'): 1, Seq('TATATGAA'): 1, Seq('ATATGAAG'): 1, Seq('TATGAAGC'): 1, Seq('ATGAAGCA'): 1, Seq('TGAAGCAA'): 1, Seq('GAAGCAAT'): 1, Seq('AAGCAATT'): 1, Seq('AGCAATTG'): 1, Seq('GCAATTGA'): 1, Seq('CAATTGAG'): 1, Seq('AATTGAGG'): 1, Seq('ATTGAGGA'): 1, Seq('TTGAGGAG'): 1, Seq('TGAGGAGT'): 1, Seq('GAGGAGTG'): 1, Seq('AGGAGTGC'): 1, Seq('GGAGTGCC'): 1, Seq('GAGTGCCT'): 1, Seq('AGTGCCTG'): 1, Seq('GTGCCTGA'): 1, Seq('TGCCTGAT'): 1, Seq('GCCTGATT'): 1, Seq('CCTGATTA'): 1, Seq('CTGATTAA'): 1, Seq('TGATTAAC'): 1, Seq('GATTAACG'): 1, Seq('ATTAACGA'): 1, Seq('TTAACGAT'): 1, Seq('TAACGATC'): 1, Seq('AACGATCC'): 1, Seq('ACGATCCC'): 1, Seq('CGATCCCT'): 1, Seq('GATCCCTG'): 1, Seq('ATCCCTGG'): 1, Seq('TCCCTGGG'): 1, Seq('CCCTGGGT'): 1, Seq('CCTGGGTT'): 1, Seq('CTGGGTTT'): 1, Seq('TGGGTTTT'): 1, Seq('GGGTTTTG'): 1, Seq('GGTTTTGC'): 1, Seq('GTTTTGCT'): 1, Seq('TTTTGCTT'): 1, Seq('TTTGCTTA'): 1, Seq('TTGCTTAA'): 1, Seq('TGCTTAAT'): 1, Seq('GCTTAATG'): 1, Seq('CTTAATGC'): 1, Seq('TTAATGCA'): 1, Seq('TAATGCAT'): 1, Seq('TGCATCTT'): 1, Seq('GCATCTTG'): 1, Seq('CATCTTGG'): 1, Seq('ATCTTGGT'): 1, Seq('TCTTGGTT'): 1, Seq('CTTGGTTC'): 1, Seq('TTGGTTCA'): 1, Seq('TGGTTCAA'): 1, Seq('GGTTCAAC'): 1, Seq('GTTCAACT'): 1, Seq('TTCAACTC'): 1, Seq('TCAACTCC'): 1, Seq('CAACTCCT'): 1, Seq('AACTCCTT'): 1, Seq('ACTCCTTC'): 1, Seq('CTCCTTCC'): 1, Seq('TCCTTCCT'): 1, Seq('CCTTCCTC'): 1, Seq('CTTCCTCA'): 1, Seq('TTCCTCAC'): 1, Seq('TCCTCACA'): 1, Seq('CCTCACAC'): 1, Seq('CTCACACA'): 1, Seq('TCACACAT'): 1, Seq('CACACATG'): 1, Seq('ACACATGC'): 1, Seq('CACATGCA'): 1, Seq('ACATGCAC'): 1, Seq('CATGCACT'): 1, Seq('ATGCACTG'): 1, Seq('TGCACTGA'): 1, Seq('GCACTGAA'): 1, Seq('CACTGAAA'): 1, Seq('ACTGAAAT'): 1, Seq('CTGAAATA'): 1, Seq('TGAAATAG'): 1})\n"
     ]
    }
   ],
   "source": [
    "# Process the Genome sequence\n",
    "from collections import defaultdict\n",
    "\n",
    "def kmer_count(sequence, k=8):\n",
    "    kmer_dict = defaultdict(int)\n",
    "    for i in range(len(sequence) - k + 1):\n",
    "        kmer = sequence[i:i+k]\n",
    "        kmer_dict[kmer] += 1\n",
    "    return kmer_dict\n",
    "\n",
    "print(kmer_count(sequences[0].seq, k=8))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "GC Content Score: 44.07252440725244\n"
     ]
    }
   ],
   "source": [
    "# GC content score\n",
    "def gc_content(seq):\n",
    "    gc_count = seq.count('G') + seq.count('C')\n",
    "    return (gc_count / len(seq)) * 100\n",
    "\n",
    "\n",
    "gc_score = gc_content(sequences[0].seq)\n",
    "print(\"GC Content Score:\", gc_score)"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "dev",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.15"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 2
}
